---
title: "i4_swab_metagenomics.Rmd"
author: "Braden T Tierney"
date: '2022-10-16'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

### METHOLODOGICAL NOTE

F2 -- HIGH COVERAGE + TOP GENES
F3/4 TOP -- HIGH COVERAGE AND LP SIGS
F3/4 BOTTOM -- HIGH COVERAGE AND NON LP SIGS WITH HIGH L2FC
F6 -- NON LP SIGS HIGH COVERAGE

####### LIBRARIES
```{r}
library(tidyverse)
library(ggplot2)
library(umap)
library(lme4)
library(broom.mixed)
library(gridExtra)
library(broom)
library(ape)
library(cowplot)
library(reshape2)
library(taxonomizr)
library(circlize)
library(ggtree)
library(RColorBrewer)
library(ComplexHeatmap)
library(ggbeeswarm)
library(lmerTest)
library(ComplexUpset)
```
####### DATA LOAD AND ALPHA DIVERSITY
```{r}

setwd('~/Dropbox (Mason Lab)/i4/i4_data_packet/')

sanitize_sample_names <- function(data){
  temp = data%>% rownames_to_column('temp') %>% mutate(namelengths = nchar(temp))
  temp = temp %>% mutate(temp = if_else(namelengths>=3 & str_sub(temp,nchar(temp),nchar(temp))=='D',str_sub(temp,1,nchar(temp)-1),temp))
  return(temp %>% column_to_rownames('temp') %>% select(-namelengths))
}

# load metadata
meta = read.csv('../i4_swab_metadata.csv') %>% mutate(location = if_else(Crew.ID == 'Capsule','Capsule',Body.Location))
meta$SeqID = gsub('ELMB_','',meta$SeqID)
meta$SeqID = gsub('SW_','',meta$SeqID)
meta$Timepoint_Recode = factor(meta$Timepoint)
levels(meta$Timepoint_Recode) = c(NA,'PRE-LAUNCH','POST-LAUNCH','PRE-LAUNCH','MID-FLIGHT','MID-FLIGHT','POST-LAUNCH','POST-LAUNCH','PRE-LAUNCH')

meta = meta %>% distinct %>% mutate(Timepoint_Recode2 = if_else(as.character(Timepoint_Recode) == 'MID-FLIGHT',Timepoint,as.character(Timepoint_Recode)))
meta$Timepoint_Recode2 = factor(meta$Timepoint_Recode2,levels = c('PRE-LAUNCH','Flight 1','Flight 2','POST-LAUNCH'))
meta$Timepoint = factor(meta$Timepoint,levels=c('21-Jun','21-Aug','Sept Pre-Launch','Flight 1','Flight 2','Sept Post-Return','November','21-Dec',NA))
meta$Timepoint_Numeric = as.numeric(meta$Timepoint)
meta$Timepoint_Recode = factor(meta$Timepoint_Recode,levels = c('MID-FLIGHT','PRE-LAUNCH','POST-LAUNCH'))

#meta = meta %>% filter(Body.Location != "Swab Water",location!='Capsule',Body.Location != 'Open Air Control', Body.Location != "Deltoid - Pre-Biospy", !is.na(Body.Location))
meta = meta %>% mutate(isoral = if_else(Body.Location == 'Oral',1,0))
meta = meta %>% mutate(isnasal = if_else(Body.Location == 'Nasal',1,0))
meta = meta %>% mutate(isstool = if_else(Body.Location == 'Stool',1,0))
meta = meta %>% mutate(isskin = if_else(Body.Location != 'Oral' & Body.Location != 'Nasal'& Body.Location != 'Stool',1,0))
meta = meta %>% mutate(Armpit = if_else(Body.Location == 'Armpit',1,0))
meta = meta %>% mutate(web = if_else(Body.Location == 'Toe Web Space',1,0))
meta = meta %>% mutate(nape = if_else(Body.Location == 'Nape of Neck',1,0))
meta = meta %>% mutate(postauric = if_else(Body.Location == 'Post-Auricular',1,0))
meta = meta %>% mutate(fore = if_else(Body.Location == 'Forearm',1,0))
meta = meta %>% mutate(bb = if_else(Body.Location == 'Belly Button',1,0))
meta = meta %>% mutate(gc = if_else(Body.Location == 'Gluteal Crease',1,0))
meta = meta %>% mutate(nasal = if_else(Body.Location == 'Nasal',1,0))
meta = meta %>% mutate(Tzone = if_else(Body.Location == 'T-Zone',1,0))
meta = meta %>% mutate(Oral = if_else(Body.Location == 'Oral',1,0))

# kraken metag
kraken_metag = readRDS('~/Dropbox (Mason Lab)/i4/i4_data_packet/revisions/kraken/bac-vir-fung_kraken2_conf-mask_species_metagenomics_decontam.rds')

# kraken metat
kraken_metat = readRDS('~/Dropbox (Mason Lab)/i4/i4_data_packet/revisions/kraken/bac-vir-fung_kraken2_conf-mask_species_metatranscriptomics_decontam.rds')

# bac GTDB metag/metat
gtdb_metag = readRDS("~/Dropbox (Mason Lab)/i4/i4_data_packet/revisions/gtdb/bacteria_xtree_05-0025_s___metagenomics_decontam.rds")
gtdb_metat = readRDS('~/Dropbox (Mason Lab)/i4/i4_data_packet/revisions/gtdb/bacteria_xtree_005-0025_s___metatranscriptomics_decontam.rds') 

# bac assembled metag/metat
#bass_metag = read.table("~/Dropbox (Mason Lab)/i4/i4_data_packet/revisions/bacterial_MAG/",check.names=F,sep='\t')
#bass_metat = read.table('ASSEMBLED-BACTERIAL-MAGS_.01_.005_metatranscriptomics_species_ra.tsv',check.names=F,sep='\t')

#bass_metag = sanitize_sample_names(bass_metag %>% t %>% as.data.frame(check.names=F))

# viral metag/metat
viral_metag = readRDS("~/Dropbox (Mason Lab)/i4/i4_data_packet/revisions/genbank/virus_xtree_01-005_g___metagenomics_decontam.rds")
viral_metat = readRDS('~/Dropbox (Mason Lab)/i4/i4_data_packet/revisions/genbank/virus_xtree_01-005_g___metatranscriptomics_decontam.rds')

metaph_metag = readRDS("~/Dropbox (Mason Lab)/i4/i4_data_packet/revisions/metaphlan4/bacteria_metaphlan4_default_s__metagenomics_decontam.rds")
metaph_metat = readRDS("~/Dropbox (Mason Lab)/i4/i4_data_packet/revisions/metaphlan4/bacteria_metaphlan4_default_s__metatranscriptomics_decontam.rds")

# viral assembled metag/metat
#viral_ass_metag = read.table("ASSEMBLED-VIRAL-GENOMES_.01_.0.005_metagenomics_species_ra.tsv",check.names=F,sep='\t')
#viral_ass_metat = read.table('ASSEMBLED-VIRAL-GENOMES_.01_.0.005_metatranscriptomics_species_ra.tsv',check.names=F,sep='\t')

#viral_ass_metag = sanitize_sample_names(viral_ass_metag %>% t %>% as.data.frame(check.names=F))

# Gene metag/metat

#### NEED TO UPDATE TO LOAD ONLY SIGNIFICANT GENES
gene_metag = readRDS('~/Dropbox (Mason Lab)/i4/i4_data_packet/gene_catalog/gene_catalog90_filtered_relab_metagenomics_subset.rds')
gene_metat = readRDS('~/Dropbox (Mason Lab)/i4/i4_data_packet/gene_catalog/gene_catalog90_filtered_relab_metatranscriptomics_subset.rds')

compute_diversity <- function(data){
  simpson = map(data, function(x) vegan::diversity(x,'simpson')) %>% data.frame(check.names=F) %>% t %>% data.frame(check.names=F) %>% rownames_to_column('sample_id') %>% mutate(type = 'SIMPSON')

  shannon = map(data, function(x) vegan::diversity(x,'shannon')) %>% data.frame(check.names=F) %>% t %>% data.frame(check.names=F) %>% rownames_to_column('sample_id')  %>% mutate(type = 'SHANNON')
  
  richness = map(data, function(x) sum(x!=0)) %>% data.frame(check.names=F)  %>% t %>% data.frame(check.names=F) %>% rownames_to_column('sample_id')  %>% mutate(type = 'RICHNESS')
  
  divs = bind_rows(shannon,simpson,richness)
  
  colnames(divs)[2] = 'diversity'

  return(divs)
}

kraken2_metag_divs = compute_diversity(kraken_metag %>% t%>% as.data.frame(check.names=F)) %>% mutate(dset = 'KRAKEN2') %>% mutate(seqtype = 'METAGENOMICS')
kraken2_metat_divs = compute_diversity(kraken_metat %>% t%>% as.data.frame(check.names=F)) %>% mutate(dset = 'KRAKEN2') %>% mutate(seqtype = 'METATRANSCRIPTOMICS')

gtdb_metag_divs = compute_diversity(gtdb_metag %>% t %>% as.data.frame(check.names=F)) %>% mutate(dset = 'GTDB')%>% mutate(seqtype = 'METAGENOMICS')
gtdb_metat_divs = compute_diversity(gtdb_metat %>% t %>% as.data.frame(check.names=F)) %>% mutate(dset = 'GTDB')%>% mutate(seqtype = 'METATRANSCRIPTOMICS')

genvir_metag_divs = compute_diversity(viral_metag %>% t %>% as.data.frame(check.names=F)) %>% mutate(dset = 'GENBANK-VIRUSES') %>% mutate(seqtype = 'METAGENOMICS')
genvir_metat_divs = compute_diversity(viral_metat) %>% mutate(dset = 'GENBANK-VIRUSES')%>% mutate(seqtype = 'METATRANSCRIPTOMICS')

#bacass_metag_divs = compute_diversity(bass_metag %>%t%>% as.data.frame(check.names=F)) %>% mutate(dset = 'ASSEMBLED-BACTERIA')%>% mutate(seqtype = 'METAGENOMICS')
#bacass_metat_divs = compute_diversity(bass_metat) %>% mutate(dset = 'ASSEMBLED-BACTERIA')%>% mutate(seqtype = 'METATRANSCRIPTOMICS')

#virass_metag_divs = compute_diversity(viral_ass_metag%>%t %>% as.data.frame(check.names=F)) %>% mutate(dset = 'ASSEMBLED-VIRUSES')%>% mutate(seqtype = 'METAGENOMICS')
#virass_metat_divs = compute_diversity(viral_ass_metat) %>% mutate(dset = 'ASSEMBLED-VIRUSES')%>% mutate(seqtype = 'METATRANSCRIPTOMICS')

gene_metag_divs = compute_diversity(gene_metag%>%  t %>% as.data.frame(check.names=F)) %>% mutate(dset = 'GENE-CATALOG')%>% mutate(seqtype = 'METAGENOMICS')
gene_metat_divs = compute_diversity(gene_metat %>% t %>% as.data.frame(check.names=F)) %>% mutate(dset = 'GENE-CATALOG')%>% mutate(seqtype = 'METATRANSCRIPTOMICS')

diversities = bind_rows(gtdb_metag_divs,gtdb_metat_divs,gene_metag_divs,gene_metat_divs)

```
####### BETA DIVERSITY COMPUTATION
```{r}
# beta diversity
compute_beta_div <- function(data,seqtype){
  out1 = vegan::vegdist(data) %>% as.matrix %>% reshape2::melt() %>% mutate(seqtype = seqtype)
  colnames(out1) = c('REF','QUERY','DISTANCE','SEQTYPE')
  return(out1)
}

#kraken_metag_betadiv = compute_beta_div(kraken_metag,'METAGENOMICS')%>% mutate(dset = 'KRAKEN2')
#kraken_metat_betadiv = compute_beta_div(kraken_metat,'METATRANSCRIPTOMICS')%>% mutate(dset = 'KRAKEN2')

viral_metag_betadiv = compute_beta_div(viral_metag %>% t %>% as.data.frame(check.names=F),'METAGENOMICS')%>% mutate(dset = 'GENBANK-VIRUSES')
viral_metat_betadiv = compute_beta_div(viral_metat %>% t %>% as.data.frame(check.names=F),'METATRANSCRIPTOMICS')%>% mutate(dset = 'GENBANK-VIRUSES')
#viral_ass_metag_bdiv = compute_beta_div(viral_ass_metag,'METAGENOMICS')%>% mutate(dset = 'ASSEMBLED-VIRUSES')
#viral_ass_metat_bdiv = compute_beta_div(viral_ass_metat %>% t%>% as.data.frame(check.names=F),'METATRANSCRIPTOMICS')%>% mutate(dset = 'ASSEMBLED-VIRUSES')

#bass_metag_bdiv = compute_beta_div(bass_metag,'METAGENOMICS')%>% mutate(dset = 'ASSEMBLED-BACTERIA')
#bass_metat_bdiv = compute_beta_div(bass_metat%>% t%>% as.data.frame(check.names=F),'METATRANSCRIPTOMICS')%>% mutate(dset = 'ASSEMBLED-BACTERIA')

#gene_metag_bdiv = compute_beta_div(gene_metag %>% t %>% as.data.frame(check.names=F),'METAGENOMICS')%>% mutate(dset = 'GENE-CATALOG')
#gene_metat_bdiv = compute_beta_div(gene_metat%>% t %>% as.data.frame(check.names=F),'METATRANSCRIPTOMICS')%>% mutate(dset = 'GENE-CATALOG')

gtdb_metag_bdiv = compute_beta_div(gtdb_metag%>% t %>% as.data.frame(check.names=F),'METAGENOMICS')%>% mutate(dset = 'GTDB')
gtdb_metat_bdiv = compute_beta_div(gtdb_metat%>% t %>% as.data.frame(check.names=F),'METATRANSCRIPTOMICS') %>% mutate(dset = 'GTDB')

metaphlan_metag_bdiv = compute_beta_div(metaph_metag%>% t %>% as.data.frame(check.names=F),'METAGENOMICS')%>% mutate(dset = 'METAPHLAN4')
metaphlan_metat_bdiv = compute_beta_div(metaph_metat%>% t %>% as.data.frame(check.names=F),'METATRANSCRIPTOMICS') %>% mutate(dset = 'METAPHLAN4')

betadivs = bind_rows(metaphlan_metag_bdiv,metaphlan_metat_bdiv,gtdb_metag_bdiv,gtdb_metat_bdiv,viral_metag_betadiv,viral_metat_betadiv)
```
####### LOAD IN TAXONOMIC AND FUNCTIONAL ANNOTATIONS
```{r}
### load in the annotation data
setwd('~/Dropbox (Mason Lab)/i4/i4_data_packet/')

# kraken taxonomies
#kraken_tax = bind_rows(kraken_tax1,kraken_tax2) %>% distinct
#taxa = getTaxonomy(kraken_tax$taxonomy_id)
#taxa = data.frame(ftaxa) %>% rownames_to_column('taxid')
#taxa$taxid = as.numeric(taxa$taxid)
#kraken_tax = inner_join(taxa,kraken_tax,by=c('taxid'='taxonomy_id')) %>% rename(kraken_column_name = name)

# genbank viral phyla
genvir_tax = read.delim('../genbank_viral_taxonomies.tsv',header=T) 
xtreemap = read.delim('../xtree_all_db_mapping',sep='\t',header=F)
genvir_tax = left_join(genvir_tax,xtreemap,by= c('contigid'='V1'))
genvir_tax = genvir_tax %>% select(V2,Phy_in_root,Cla_in_root,Ord_in_root,Fam_in_root,Gen_in_root,Spe_in_root) %>% rename(query= V2)
colnames(genvir_tax) = gsub('_in_root','',colnames(genvir_tax))

gtdb_metag_t = gtdb_metag %>% t %>% as.data.frame(check.names=F)
gtdb_metat_t = gtdb_metat %>% t %>% as.data.frame(check.names=F)
# gtdb taxonomies (parse from file itself)
gtdb_tax = separate(col=`c(colnames(gtdb_metag_t), colnames(gtdb_metat_t))`,tibble(c(colnames(gtdb_metag_t),colnames(gtdb_metat_t))),sep=';',into = c('Domain','Phylum','Class','Order','Family','Genus','Species'))
gtdb_tax = gtdb_tax %>% mutate(Domain = str_replace(Domain, "d__", ""))
gtdb_tax = gtdb_tax %>% mutate(Phylum = str_replace(Phylum, "p__", ""))
gtdb_tax = gtdb_tax %>% mutate(Class = str_replace(Class, "c__", ""))
gtdb_tax = gtdb_tax %>% mutate(Order = str_replace(Order, "o__", ""))
gtdb_tax = gtdb_tax %>% mutate(Family = str_replace(Family, "f__", ""))
gtdb_tax = gtdb_tax %>% mutate(Genus = str_replace(Genus, "g__", ""))
gtdb_tax = gtdb_tax %>% mutate(Species = str_replace(Species, "s__", ""))
gtdb_tax = bind_cols(tibble(c(colnames(gtdb_metag_t),colnames(gtdb_metat_t))),gtdb_tax)
colnames(gtdb_tax)[1] = 'query'

metaph_metag_t = metaph_metag %>% t %>% as.data.frame(check.names=F)
metaph_metat_t = metaph_metat %>% t %>% as.data.frame(check.names=F)
# metaph taxonomies (parse from file itself)
metaph_tax = separate(col=`c(colnames(metaph_metag_t), colnames(metaph_metat_t))`,tibble(c(colnames(metaph_metag_t),colnames(metaph_metat_t))),sep='\\|',into = c('Domain','Phylum','Class','Order','Family','Genus','Species'))
metaph_tax = metaph_tax %>% mutate(Domain = str_replace(Domain, "d__", ""))
metaph_tax = metaph_tax %>% mutate(Phylum = str_replace(Phylum, "p__", ""))
metaph_tax = metaph_tax %>% mutate(Class = str_replace(Class, "c__", ""))
metaph_tax = metaph_tax %>% mutate(Order = str_replace(Order, "o__", ""))
metaph_tax = metaph_tax %>% mutate(Family = str_replace(Family, "f__", ""))
metaph_tax = metaph_tax %>% mutate(Genus = str_replace(Genus, "g__", ""))
metaph_tax = metaph_tax %>% mutate(Species = str_replace(Species, "s__", ""))
metaph_tax = bind_cols(tibble(c(colnames(metaph_metag_t),colnames(metaph_metat_t))),metaph_tax)
colnames(metaph_tax)[1] = 'query'


# assembled bacterial taxonomies
taxdata = read.csv('~/Dropbox (Mason Lab)/i4/i4_data_packet/gtdbtk.bac120.summary.tsv',sep='\t',row.names = 1) %>% mutate(phylum = strsplit(classification,'p__') %>% map_chr(2) %>% strsplit(';') %>% map_chr(1)) %>% rownames_to_column('id') 
taxdata$species = map(taxdata$classification, function(x) gsub("^.*__", "", x)) %>% unlist %>% unname

bar = taxdata$classification
bar = gsub('s__','',bar)
bar = gsub('g__','',bar)
bar = gsub('f__','',bar)
bar = gsub('c__','',bar)
bar = gsub('o__','',bar)
bar = gsub('p__','',bar)
bar = gsub('d__','',bar)

foo = str_split_fixed(bar, ";", 8) %>% data.frame

taxdata = bind_cols(taxdata,foo)
bacass_taxa = taxdata %>% mutate(labelval = if_else(X7 != '',X7,if_else(X6 != '',X6,if_else(X5 != '',X5,if_else(X4 != '',X4,if_else(X3 != '',X3, if_else(X2 != '',X2,X1))))))) %>% select(id,X1,X2,X3,X4,X5,X6,X7,X8) %>% rename(Domain = X1,Phylum = X2, Class = X3, Order = X4,Family = X5,Genus = X6,Species = X7,query = id)

# assembled viral taxonomies
taxdata = read.csv('closest_genbank_taxonomies.csv',sep=',',row.names = 1) %>% mutate(phylum = strsplit(taxonomy,';') %>% map_chr(1)) %>% group_by(query) %>% slice_min(distance_from_reference,n=1) %>% ungroup %>%  select(query,phylum,taxonomy)
taxdata$species = taxdata$taxonomy %>% map(function(x) strsplit(stringi::stri_reverse(x),';') %>% map_chr(1) %>% stringi::stri_reverse()) %>% unlist %>% unname
virass_tax = taxdata

# gene function mapping
gene_annos = read.csv('~/Dropbox (Mason Lab)/i4/i4_data_packet/gene_catalog/i4_90perc_congene_annotations.csv',header=T) 
gene_annos = gene_annos %>% select(Locus.Tag,Gene,Product,DbXrefs)
colnames(gene_annos) = c('yvar','gene_name','product','other_annotations')
```
####### DEPRECATED -------- ALPHA DIVERSITY 1
```{r}
### alpha diversity analysis

metasub = meta %>% filter(location != 'Capsule',Body.Location != "Swab Water", Body.Location != "Deltoid - Pre-Biospy", !is.na(Body.Location))

metasub$Timepoint_Recode = factor(metasub$Timepoint_Recode,levels = c('PRE-LAUNCH','MID-FLIGHT','POST-LAUNCH'))
metasub = metasub %>% mutate(Timepoint = if_else(Timepoint == 'Flight 1' | Timepoint == 'Flight 2','Mid-Flight',as.character(Timepoint)))
metasub$Timepoint = factor(metasub$Timepoint,levels=c('Mid-Flight','21-Jun','21-Aug','Sept Pre-Launch','Sept Post-Return','November','21-Dec',NA))
metasub$Timepoint_Recode = factor(metasub$Timepoint_Recode,levels=c('MID-FLIGHT','PRE-LAUNCH','POST-LAUNCH'))

diversity_meta = inner_join(diversities,metasub,by=c('sample_id'='SeqID')) %>% mutate(skinoralnasal = if_else(isoral == 1,'ORAL',if_else(isskin==1,'SKIN','NASAL')))

compute_overall_div_lmers <- function(dataset,dset2){
  a = lmer(data = dataset %>% filter(type == 'SHANNON',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode + (1|Crew.ID)) %>% tidy %>% mutate(type = 'SHANNON',seqtype = 'METAGENOMICS',dset = dset2)
  b = lmer(data = dataset %>% filter(type == 'SIMPSON',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode + (1|Crew.ID)) %>% tidy%>% mutate(type = 'SIMPSON',seqtype = 'METAGENOMICS',dset = dset2) 
  c = lmer(data = dataset %>% filter(type == 'RICHNESS',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METAGENOMICS',dset = dset2)
  d = lmer(data = dataset %>% filter(type == 'SHANNON',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode + (1|Crew.ID)) %>% tidy %>% mutate(type = 'SHANNON',seqtype = 'METATRANSCRIPTOMICS',dset = dset2)
  e = lmer(data = dataset %>% filter(type == 'SIMPSON',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode + (1|Crew.ID)) %>% tidy%>% mutate(type = 'SIMPSON',seqtype = 'METATRANSCRIPTOMICS',dset = dset2) 
  f = lmer(data = dataset %>% filter(type == 'RICHNESS',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METATRANSCRIPTOMICS',dset = dset2)
  g = lmer(data = dataset %>% filter(type == 'RICHNESS',dset == dset2), diversity ~ Timepoint_Recode*seqtype + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METATRANSCRIPTOMICS VS METAGENOMICS',dset = dset2)
  h = lmer(data = dataset %>% filter(type == 'SIMPSON',dset == dset2), diversity ~ Timepoint_Recode*seqtype + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METATRANSCRIPTOMICS VS METAGENOMICS',dset = dset2)
  i = lmer(data = dataset %>% filter(type == 'SHANNON',dset == dset2), diversity ~ Timepoint_Recode*seqtype + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METATRANSCRIPTOMICS VS METAGENOMICS',dset = dset2)
  return(bind_rows(a,b,c,d,e,f,g,h,i))
}


compute_overall_div_lmers_bs <- function(dataset,dset2){
  a = lmer(data = dataset %>% filter(type == 'SHANNON',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode*isoral + (1|Crew.ID)) %>% tidy %>% mutate(type = 'SHANNON',seqtype = 'METAGENOMICS',dset = dset2)%>% mutate(site = 'ORAL')
  b = lmer(data = dataset %>% filter(type == 'SIMPSON',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode*isskin + (1|Crew.ID)) %>% tidy%>% mutate(type = 'SIMPSON',seqtype = 'METAGENOMICS',dset = dset2)  %>% mutate(site = 'SKIN')
  c = lmer(data = dataset %>% filter(type == 'RICHNESS',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode*isnasal + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METAGENOMICS',dset = dset2)%>% mutate(site = 'NASAL')
  
  d = lmer(data = dataset %>% filter(type == 'SHANNON',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode*isoral + (1|Crew.ID)) %>% tidy %>% mutate(type = 'SHANNON',seqtype = 'METAGENOMICS',dset = dset2)%>% mutate(site = 'ORAL')
  e = lmer(data = dataset %>% filter(type == 'SIMPSON',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode*isskin + (1|Crew.ID)) %>% tidy%>% mutate(type = 'SIMPSON',seqtype = 'METAGENOMICS',dset = dset2) %>% mutate(site = 'SKIN')
  f = lmer(data = dataset %>% filter(type == 'RICHNESS',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode*isnasal + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METAGENOMICS',dset = dset2)%>% mutate(site = 'NASAL')
  
  h = lmer(data = dataset %>% filter(type == 'SHANNON',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode*isoral + (1|Crew.ID)) %>% tidy %>% mutate(type = 'SHANNON',seqtype = 'METAGENOMICS',dset = dset2)%>% mutate(site = 'ORAL')
  i = lmer(data = dataset %>% filter(type == 'SIMPSON',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode*isskin  + (1|Crew.ID)) %>% tidy%>% mutate(type = 'SIMPSON',seqtype = 'METAGENOMICS',dset = dset2)  %>% mutate(site = 'SKIN')
  j = lmer(data = dataset %>% filter(type == 'RICHNESS',seqtype == 'METAGENOMICS',dset == dset2), diversity ~ Timepoint_Recode*isnasal + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METAGENOMICS',dset = dset2)%>% mutate(site = 'NASAL')
  
  out1 = bind_rows(a,b,c,d,e,f,h,i,j)
  
   a = lmer(data = dataset %>% filter(type == 'SHANNON',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode*isoral + (1|Crew.ID)) %>% tidy %>% mutate(type = 'SHANNON',seqtype = 'METATRANSCRIPTOMICS',dset = dset2)%>% mutate(site = 'ORAL')
  b = lmer(data = dataset %>% filter(type == 'SIMPSON',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode*isskin  + (1|Crew.ID)) %>% tidy%>% mutate(type = 'SIMPSON',seqtype = 'METATRANSCRIPTOMICS',dset = dset2)  %>% mutate(site = 'SKIN')
  c = lmer(data = dataset %>% filter(type == 'RICHNESS',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode*isnasal + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METATRANSCRIPTOMICS',dset = dset2)%>% mutate(site = 'NASAL')
  
  d = lmer(data = dataset %>% filter(type == 'SHANNON',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode*isoral + (1|Crew.ID)) %>% tidy %>% mutate(type = 'SHANNON',seqtype = 'METATRANSCRIPTOMICS',dset = dset2)%>% mutate(site = 'ORAL')
  e = lmer(data = dataset %>% filter(type == 'SIMPSON',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode*isskin + (1|Crew.ID)) %>% tidy%>% mutate(type = 'SIMPSON',seqtype = 'METATRANSCRIPTOMICS',dset = dset2)  %>% mutate(site = 'SKIN')
  f = lmer(data = dataset %>% filter(type == 'RICHNESS',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode*isnasal + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METATRANSCRIPTOMICS',dset = dset2)%>% mutate(site = 'NASAL')
  
  h = lmer(data = dataset %>% filter(type == 'SHANNON',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode*isoral + (1|Crew.ID)) %>% tidy %>% mutate(type = 'SHANNON',seqtype = 'METATRANSCRIPTOMICS',dset = dset2)%>% mutate(site = 'ORAL')
  i = lmer(data = dataset %>% filter(type == 'SIMPSON',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode*isskin + (1|Crew.ID)) %>% tidy%>% mutate(type = 'SIMPSON',seqtype = 'METATRANSCRIPTOMICS',dset = dset2) %>% mutate(site = 'SKIN')
  j = lmer(data = dataset %>% filter(type == 'RICHNESS',seqtype == 'METATRANSCRIPTOMICS',dset == dset2), diversity ~ Timepoint_Recode*isnasal + (1|Crew.ID)) %>% tidy %>% mutate(type = 'RICHNESS',seqtype = 'METATRANSCRIPTOMICS',dset = dset2) %>% mutate(site = 'NASAL')
  
  out2 = bind_rows(a,b,c,d,e,f,h,i,j)

  return(bind_rows(out1,out2))
}

# OVERALL DIVERSITY CHANGES
out=list()
for(d in diversity_meta$dset %>% unique){
  out[[d]] = compute_overall_div_lmers(diversity_meta,d)
}
out = bind_rows(out)
write.csv(out,'~/Dropbox (Mason Lab)/i4/random_tables_supp/overall_alpha_div_lmms.csv')
# BODY SITE SPECIFIC DIVERSITY CHANGES
out=list()
for(d in diversity_meta$dset %>% unique){
  out[[d]] = compute_overall_div_lmers_bs(diversity_meta,d)
}
out = bind_rows(out)
write.csv(out,'~/Dropbox (Mason Lab)/i4/random_tables_supp/overall_alpha_div_lmms_bs.csv')
setwd('~/Dropbox (Mason Lab)/i4/diversity_plots')

ggplot(data = diversity_meta %>% filter(type =='SHANNON',!is.na(skinoralnasal)),aes(x = Timepoint_Numeric,y = (diversity))) + geom_smooth(method = 'loess') + theme_bw()+ facet_wrap(~dset + skinoralnasal + seqtype,scales='free')
ggsave('shannon_forsupp.pdf',width=15,height=20)

ggplot(data = diversity_meta %>% filter(type =='SIMPSON',!is.na(skinoralnasal)),aes(x = Timepoint_Numeric,y = (diversity))) + geom_smooth(method = 'loess') + theme_bw()+ facet_wrap(~dset + skinoralnasal + seqtype,scales='free')
ggsave('simpson_forsupp.pdf',width=15,height=20)

ggplot(data = diversity_meta %>% filter(type !='RICHNESS',!is.na(skinoralnasal)),aes(x = Timepoint_Numeric,color=seqtype,y = (diversity))) + geom_smooth(method = 'loess') + theme_bw()+ facet_wrap(~dset + skinoralnasal + type,scales='free')+scale_color_manual(values = c('darkblue','#BE1E2D'))
ggsave('otherdiv_forsupp.pdf',width=15,height=20)


### FOR MAIN TEXT
diversity_meta$dset = factor(diversity_meta$dset, levels = c('GTDB','GENBANK-VIRUSES','GENE-CATALOG','KRAKEN2','ASSEMBLED-VIRUSES','ASSEMBLED-BACTERIA'))
diversity_meta$skinoralnasal = factor(diversity_meta$skinoralnasal, levels = c('ORAL','NASAL','SKIN'))

ggplot(data = diversity_meta %>% filter(!is.na(skinoralnasal),dset == 'GTDB',seqtype == 'METAGENOMICS',type != 'SIMPSON',type != 'SHANNON'),aes(x = Timepoint_Numeric,y = (diversity))) + geom_smooth(method = 'loess',color = 'blue') + theme_bw()+ facet_grid(type + skinoralnasal~ . ,scales='free_y')
ggsave('~/Dropbox (Mason Lab)/i4/diversity_plots/div_gtdb_metag.pdf',width=3,height=6)

#ggplot(data = diversity_meta %>% filter(!is.na(skinoralnasal),dset == 'GENBANK-VIRUSES',seqtype == 'METAGENOMICS',type != 'SIMPSON',type != 'SHANNON'),aes(x = Timepoint_Numeric,y = (diversity))) + geom_smooth(method = 'loess',color = 'darkblue') + theme_bw()+ facet_grid(type + skinoralnasal~ . ,scales='free_y')
#ggsave('~/Dropbox (Mason Lab)/i4/diversity_plots/div_genvir_metag.pdf',width=3,height=6)

#ggplot(data = diversity_meta %>% filter(!is.na(skinoralnasal),dset == 'GENE-CATALOG',seqtype == 'METAGENOMICS',type != 'SIMPSON',type != 'SHANNON'),aes(x = Timepoint_Numeric,y = (diversity))) + geom_smooth(method = 'loess',color = 'darkblue') + theme_bw()+ facet_grid(type + skinoralnasal~ . ,scales='free_y')
#ggsave('~/Dropbox (Mason Lab)/i4/diversity_plots/div_genecat_metag.pdf',width=3,height=6)


#ggplot(data = diversity_meta %>% filter(!is.na(skinoralnasal),dset == 'GTDB',seqtype == 'METATRANSCRIPTOMICS',type != 'SIMPSON',type != 'SHANNON'),aes(x = Timepoint_Numeric,y = (diversity))) + geom_smooth(method = 'loess',color = '#BE1E2D') + theme_bw()+ facet_grid(type + skinoralnasal~ . ,scales='free_y')
#ggsave('~/Dropbox (Mason Lab)/i4/diversity_plots/div_gtdb_metat.pdf',width=3,height=6)

#ggplot(data = diversity_meta %>% filter(!is.na(skinoralnasal),dset == 'GENBANK-VIRUSES',seqtype == 'METATRANSCRIPTOMICS',type != 'SIMPSON',type != 'SHANNON'),aes(x = Timepoint_Numeric,y = (diversity))) + geom_smooth(method = 'loess',color = '#BE1E2D') + theme_bw()+ facet_grid(type + skinoralnasal~ . ,scales='free_y')
#ggsave('~/Dropbox (Mason Lab)/i4/diversity_plots/div_genvir_metat.pdf',width=3,height=6)

#ggplot(data = diversity_meta %>% filter(!is.na(skinoralnasal),dset == 'GENE-CATALOG',seqtype == 'METATRANSCRIPTOMICS',type != 'SIMPSON',type != 'SHANNON'),aes(x = Timepoint_Numeric,y = (diversity))) + geom_smooth(method = 'loess',color = '#BE1E2D') + theme_bw()+ facet_grid(type + skinoralnasal~ . ,scales='free_y')
#ggsave('~/Dropbox (Mason Lab)/i4/diversity_plots/div_genecat_metat.pdf',width=3,height=6)

```
####### BETA DIVERSITY 1
```{r}
### beta diversity analysis (similarity with each other and capsule over time, change in different bodysites over time)

metasub = meta %>% filter(Body.Location != "Swab Water", Body.Location != "Deltoid - Pre-Biospy", !is.na(Body.Location))

metasub$Timepoint_Recode = factor(metasub$Timepoint_Recode,levels = c('PRE-LAUNCH','MID-FLIGHT','POST-LAUNCH'))
metasub = metasub %>% mutate(Timepoint = if_else(Timepoint == 'Flight 1' | Timepoint == 'Flight 2','Mid-Flight',as.character(Timepoint)))
#metasub$Timepoint = factor(metasub$Timepoint,levels=c('Mid-Flight','21-Jun','21-Aug','Sept Pre-Launch','Sept Post-Return','November','21-Dec',NA))
metasub$Timepoint_Recode = factor(metasub$Timepoint_Recode,levels=c('MID-FLIGHT','PRE-LAUNCH','POST-LAUNCH'))

m_ref = metasub
m_query = metasub

colnames(m_ref) = paste('REF_',colnames(metasub),sep='') 
m_ref = m_ref %>% dplyr::rename(REF = REF_SeqID)
colnames(m_query) = paste('QUERY_',colnames(metasub),sep='')
m_query = m_query %>% dplyr::rename(QUERY = QUERY_SeqID)

betadivs_meta = inner_join(betadivs,m_query %>% select(QUERY,QUERY_location,QUERY_Timepoint_Recode,QUERY_Timepoint_Numeric,QUERY_Timepoint,QUERY_Timepoint_Recode2,QUERY_Body.Location,QUERY_Crew.ID),by=c('QUERY'='QUERY')) 
betadivs_meta = inner_join(betadivs_meta,m_ref %>% select(REF,REF_location,REF_Timepoint_Recode,REF_Timepoint_Numeric,REF_Timepoint,REF_Timepoint_Recode2,REF_Body.Location,REF_Crew.ID),by=c('REF'='REF')) 

betadivs_meta  = betadivs_meta %>% filter( QUERY_Body.Location!='Stool',REF_Body.Location !='Stool')

```
####### BETA DIVERSITY VS CAPSULE
```{r}
# visualizing similarity to each other vs capsule
col_fun = colorRamp2(c(.1,.4,.8,1), rev(c("#2ED8F0", "#1768BD","#000000","#88979E")))

plot_betadiv_info <- function(betadivs_meta,refdb,dtype){
  betadivs_meta_sub = betadivs_meta %>% filter(dset ==refdb,SEQTYPE == dtype) %>% filter(QUERY_location == REF_location | REF_location == 'Capsule',REF_Crew.ID != QUERY_Crew.ID,QUERY_Timepoint == REF_Timepoint,REF_Timepoint_Recode == QUERY_Timepoint_Recode,QUERY_location != 'Open Air Control',REF_location != 'Open Air Control')%>% select(DISTANCE,QUERY_location,REF_location,QUERY_Timepoint_Recode,REF_Timepoint_Recode,REF_Crew.ID) 
  
  wgs_bacterial_dist_meta_sub = betadivs_meta_sub %>% group_by(REF_Crew.ID,QUERY_Timepoint_Recode,QUERY_location) %>% summarise(DISTANCE_AVG = mean(DISTANCE)) %>% dcast(QUERY_Timepoint_Recode + REF_Crew.ID ~ QUERY_location,value.var = 'DISTANCE_AVG') 
  
  a = filter(wgs_bacterial_dist_meta_sub %>% filter(QUERY_Timepoint_Recode == 'PRE-LAUNCH')) %>% select(-QUERY_Timepoint_Recode) %>% filter(REF_Crew.ID != 'Capsule')%>% column_to_rownames('REF_Crew.ID') 
  b = filter(wgs_bacterial_dist_meta_sub %>% filter(QUERY_Timepoint_Recode == 'MID-FLIGHT'))%>% select(-QUERY_Timepoint_Recode) %>% column_to_rownames('REF_Crew.ID')
  c = filter(wgs_bacterial_dist_meta_sub %>% filter(QUERY_Timepoint_Recode == 'POST-LAUNCH'))%>% select(-QUERY_Timepoint_Recode) %>% column_to_rownames('REF_Crew.ID')
  
  #roworder = c("Oral", "Gluteal Crease","Nasal","T-Zone","Post-Auricular","Belly Button","Toe Web Space","Nape of Neck", "Forearm","Armpit")
  roworder = b['Capsule',] %>% melt %>% arrange(desc(value)) %>% mutate(variable = as.character(variable))%>% select(variable) %>% unlist %>% unname
  
  betadivs_meta_sub$REF_Timepoint_Recode = factor(betadivs_meta_sub$REF_Timepoint_Recode,levels=c('PRE-LAUNCH','MID-FLIGHT','POST-LAUNCH'))
  betadivs_meta_sub$QUERY_Timepoint_Recode = factor(betadivs_meta_sub$QUERY_Timepoint_Recode,levels=c('PRE-LAUNCH','MID-FLIGHT','POST-LAUNCH'))
  betadivs_meta_sub$QUERY_location = factor(betadivs_meta_sub$QUERY_location,levels=roworder)
  overallplot = ggplot(betadivs_meta_sub %>% filter(REF_location != 'Capsule'),aes(x = QUERY_location,y=DISTANCE,color = as.factor(QUERY_Timepoint_Recode)))+ geom_quasirandom(dodge.width =.8,alpha=.3) +geom_boxplot(alpha=.1)  + theme_bw() +scale_color_brewer(palette='Set1') + xlab('') + theme(axis.text.x = element_blank()) + ylim(0,1)
  
  ### COMPUTE STATS 
  
  temp = betadivs_meta_sub %>% mutate(REF_Timepoint_Recode = factor(REF_Timepoint_Recode,levels= c('PRE-LAUNCH','MID-FLIGHT','POST-LAUNCH'))) %>% filter(REF_location != 'Capsule')
  
  regout = lmer(data = temp,DISTANCE ~ REF_Timepoint_Recode*REF_location + (1|REF_Crew.ID)) %>% tidy
  
  write.csv(regout,'~/Dropbox (Mason Lab)/i4/random_tables_supp/betadiversity_regressions.csv')
  
  ap = Heatmap(a %>% select(roworder),cluster_columns =F,name = 'Bacterial WGS Dissimilarity',column_title_side = 'top',column_dend_side = 'bottom',row_split  = c('Crew','Crew','Crew','Crew'),col=col_fun,show_row_names = T,cluster_rows=F)
  bp =  Heatmap(b %>% select(roworder),cluster_columns =F,name = 'Bacterial WGS Dissimilarity',column_title_side = 'top',column_dend_side = 'bottom',row_split  = c('Crew','Crew','Crew','Crew','Capsule'),col=col_fun,show_row_names = T,cluster_rows=F)
  cp = Heatmap(c %>% select(roworder),cluster_columns =F,name = 'Bacterial WGS Dissimilarity',column_title_side = 'top',column_dend_side = 'bottom',row_split  = c('Crew','Crew','Crew','Crew'),col=col_fun,show_row_names = T,cluster_rows=F)
  
  heatmapplot = ap %v% bp %v% cp
  
  ggsave(plot = overallplot,filename = paste('beta_diversity/overview_',refdb,'_',dtype,'.pdf',sep=''),width=8,height=2)
  pdf(paste('beta_diversity/',refdb,'_',dtype,'.pdf',sep=''),width=6,height=4)
  draw(heatmapplot)
  dev.off()
  
  return(list(overallplot,heatmapplot,regout))  
}

setwd('~/Dropbox (Mason Lab)/i4')

gtdb_metag_bdivplots = plot_betadiv_info(betadivs_meta,'GTDB','METAGENOMICS')
gtdb_metat_bdivplots = plot_betadiv_info(betadivs_meta,'GTDB','METATRANSCRIPTOMICS')

#kraken_metag_bdivplots = plot_betadiv_info(betadivs_meta,'KRAKEN2','METAGENOMICS')
#kraken_metat_bdivplots = plot_betadiv_info(betadivs_meta,'KRAKEN2','METATRANSCRIPTOMICS')

#bacass_metag_bdivplots = plot_betadiv_info(betadivs_meta,'ASSEMBLED-BACTERIA','METAGENOMICS')
#bacass_metat_bdivplots = plot_betadiv_info(betadivs_meta,'ASSEMBLED-BACTERIA','METATRANSCRIPTOMICS')

#virass_metag_bdivplots = plot_betadiv_info(betadivs_meta,'ASSEMBLED-VIRUSES','METAGENOMICS')
#virass_metat_bdivplots = plot_betadiv_info(betadivs_meta,'ASSEMBLED-VIRUSES','METATRANSCRIPTOMICS')

virgen_metag_bdivplots = plot_betadiv_info(betadivs_meta,'GENBANK-VIRUSES','METAGENOMICS')
virgen_metat_bdivplots = plot_betadiv_info(betadivs_meta,'GENBANK-VIRUSES','METATRANSCRIPTOMICS')

metaph_metag_bdivplots = plot_betadiv_info(betadivs_meta,'METAPHLAN4','METAGENOMICS')
metaph_metat_bdivplots = plot_betadiv_info(betadivs_meta,'METAPHLAN4','METATRANSCRIPTOMICS')

#gene_metag_bdivplots = plot_betadiv_info(betadivs_meta,'GENE-CATALOG','METAGENOMICS')
#gene_metat_bdivplots = plot_betadiv_info(betadivs_meta,'GENE-CATALOG','METATRANSCRIPTOMICS')

betadivreg = bind_rows(bind_rows(metaph_metag_bdivplots[[3]] %>% mutate(seqtype = 'METAGENOMICS'),metaph_metag_bdivplots[[3]] %>% mutate(seqtype = 'METATRANSCRIPTOMICS'),gtdb_metag_bdivplots[[3]] %>% mutate(seqtype = 'METAGENOMICS'),gtdb_metat_bdivplots[[3]]%>% mutate(seqtype = 'METATRANSCRIPTOMICS')) %>% mutate(dataset = 'GTDB'),bind_rows(virgen_metag_bdivplots[[3]] %>% mutate(seqtype = 'METAGENOMICS'),virgen_metat_bdivplots[[3]]%>% mutate(seqtype = 'METATRANSCRIPTOMICS')) %>% mutate(dataset = 'GENBANK VIRUSES'))
write.csv(betadivreg,'~/Dropbox (Mason Lab)/i4/random_tables_supp/betadivreg.csv')


```
######## SPLINES AND SUMMARY PLOTS
```{r}
### GENERATE SUMMARY PLOTS

# get the regression data
fortrendline = read.csv('~/Dropbox (Mason Lab)/i4/revisions/regressionoutput/parsed/regression_data_timetrends_species_decontam.csv') %>% filter(dset != 'GENBANK-VIRUSES',dset != 'PHANTA')
# get the regression data genbank virus
fortrendline3 = read.csv('~/Dropbox (Mason Lab)/i4/revisions/regressionoutput/parsed/regression_data_timetrends_genus_decontam.csv')  %>% filter(dset == 'GENBANK-VIRUSES' |dset == 'PHANTA') 
fortrendline = bind_rows(fortrendline,fortrendline3)

regression_output = readRDS('~/Dropbox (Mason Lab)/i4/revisions/regressionoutput/parsed/regression_data_timetrends_species_decontam.rds')%>% filter(dset != 'GENBANK-VIRUSES',dset != 'PHANTA')
regression_output2 = readRDS('~/Dropbox (Mason Lab)/i4/revisions/regressionoutput/parsed/regression_data_timetrends_genus_decontam.rds') %>% filter(dset == 'GENBANK-VIRUSES' |dset == 'PHANTA') 
regression_output = bind_rows(regression_output,regression_output2)

setwd('~/Dropbox (Mason Lab)/i4/')

simple_splines <- function(dataset,metagdata,metatdata,fortrendline){
	temp_metag = fortrendline %>% filter(dset == dataset,seqtype == 'METAGENOMICS',regclass2 != 'Skin')
	temp_metat = fortrendline %>% filter(dset == dataset,seqtype == 'METATRANSCRIPTOMICS',regclass2 != 'Skin')

	temp2_metag_sub = metagdata%>% t %>% as.data.frame(check.names=F)%>% select(any_of(temp_metag$yvar)) %>% t %>% as.data.frame(check.names = F) %>% rownames_to_column('yvar') %>% melt
	temp_metag = inner_join(temp_metag,temp2_metag_sub,relationship = "many-to-many")
	temp_metag = left_join(temp_metag,meta %>% rename(variable = SeqID) %>% filter(!grepl('SW_',Prefix)))

	temp2_metat_sub = metatdata%>% t %>% as.data.frame(check.names=F)%>% select(any_of(temp_metat$yvar)) %>% t %>% as.data.frame(check.names = F) %>% rownames_to_column('yvar') %>% melt
	temp_metat = inner_join(temp_metat,temp2_metat_sub,relationship = "many-to-many")
	temp_metat = left_join(temp_metat,meta %>% rename(variable = SeqID)%>% filter(!grepl('SW_',Prefix)))

	temp = bind_rows(temp_metag,temp_metat)
	temp = temp %>% mutate(mergeid = paste(yvar,dset,seqtype,location))
	temp = temp %>% filter(mergeid %in% fortrendline$mergeid)

	## compute log2fc -- BEFORE/AFTER, BEFORE MID, AFTER
	l2fc = temp %>% select(yvar,Timepoint_Recode,location,seqtype,dset,value) %>% group_by(yvar,Timepoint_Recode,location,seqtype,dset) %>% filter(!is.na(Timepoint_Recode))%>% summarise(mean = mean(value)) %>% dcast(yvar+location+seqtype+dset~Timepoint_Recode,value.var='mean')
	l2fc[is.na(l2fc)]=0
	l2fc$PRE_MID = log2((l2fc$`PRE-LAUNCH`+0.0000001)/(l2fc$`MID-FLIGHT`+0.0000001))
	l2fc$POST_MID = log2((l2fc$`POST-LAUNCH`+0.0000001)/(l2fc$`MID-FLIGHT`+0.0000001))
	l2fc$POST_PRE = log2((l2fc$`POST-LAUNCH`+0.0000001)/(l2fc$`PRE-LAUNCH`+0.0000001))

	#q1=quantile(abs(l2fc$PRE_MID),na.rm=T,.25)[[1]]
	#q2=quantile(abs(l2fc$POST_MID),na.rm=T,.25)[[1]]
	#q3=quantile(abs(l2fc$POST_PRE),na.rm=T,.25)[[1]]

	#l2fc = l2fc %>% filter(PRE_MID > q1 | PRE_MID > q2 |PRE_MID > q3)

	l2fc = l2fc %>% select(-all_of(c("MID-FLIGHT","PRE-LAUNCH","POST-LAUNCH"))) # %>% melt(id.vars = c('yvar','location','seqtype','dset'))
	colnames(l2fc) = c('yvar','location','seqtype','dset','PRE_MID','POST_MID','POST_PRE')
	temp = inner_join(temp,l2fc)

	return(list(temp %>% mutate(PLOTTYPE = dataset) %>% filter(regclass2 == location),plot))
}

# filter to viral subtypes
annovir = read.csv('~/Dropbox (Mason Lab)/i4/viral_phylum_for_annotations.csv') %>% mutate(phylum2 = gsub('p__','',phylum))

# LOAD IN VIRAL TAXONOMIC MAPPING
mapper = read.delim('~/Dropbox (Mason Lab)/i4/genbank_viral_taxonomies.tsv',sep='\t') %>% select(Phy_in_root,Gen_in_root,Spe_in_root) %>% mutate(Phy_in_root = paste0('p__','',Phy_in_root))
colnames(mapper) = c('phylum','genus','species')
annovir = inner_join(mapper,annovir)

#fortrendline = fortrendline %>% mutate(species = strsplit(yvar,';s__') %>% map_chr())

fullcounts = fortrendline %>% filter(dset == 'GTDB' | dset == 'GENBANK-VIRUSES' | dset == 'GENE-CATALOG', regclass2 != 'Skin') %>% select(dset,seqtype,timetrend) %>% group_by(dset,seqtype,timetrend) %>% count
fullcounts$dset = factor(fullcounts$dset,levels = c('ASSEMBLED-BACTERIA','METAPHLAN4','GTDB','ASSEMBLED-VIRUSES','GENBANK-VIRUSES','PHANTA-VIRUSES','GENE-CATALOG',"KRAKEN2-CONF0.2-MASK","KRAKEN2-CONF0.0-MASK","KRAKEN2-CONF0.2-NOMASK","KRAKEN2-CONF0.0-NOMASK"))

all_combinations <- expand_grid(timetrend = unique(fullcounts$timetrend),seqtype = unique(fullcounts$seqtype),dset = unique(fullcounts$dset))

fullcounts_expanded <- left_join(all_combinations, fullcounts, by = c("timetrend", "seqtype", "dset")) %>%replace_na(list(n = 0))

fullcounts_expanded$dset = as.character(fullcounts_expanded$dset)
fullcounts_expanded$dset[fullcounts_expanded$dset == 'GTDB'] = 'Bacterial\nspecies'
fullcounts_expanded$dset[fullcounts_expanded$dset == 'GENBANK-VIRUSES'] = 'Viral genera'
fullcounts_expanded$dset[fullcounts_expanded$dset == 'GENE-CATALOG'] = 'Genes'

fullcounts_expanded$dset = factor(fullcounts_expanded$dset,levels = c('Bacterial\nspecies','Viral genera','Genes'))

ggplot(fullcounts_expanded,aes(x = timetrend, y = n+1, fill =seqtype)) + theme_bw() + geom_bar(stat='identity',position='dodge') + facet_grid(rows = vars(dset),scales='free_y',space = 'free_x') + ylab('Number of significant features') +scale_fill_manual(values = c('darkblue','#BE1E2D'))+ theme(legend.position = 'None') + theme(axis.text.x  = element_text(angle=60,hjust=1)) + scale_y_log10()
ggsave('~/Dropbox (Mason Lab)/i4/regressionsummary/overall_summarycounts.pdf',width=3,height=4)

summaryplotfodder = fortrendline %>% select(dset,seqtype,regclass2,timetrend) %>% group_by(dset,seqtype,regclass2,timetrend) %>% count

summaryplotfodder$dset = factor(summaryplotfodder$dset,levels = c('ASSEMBLED-BACTERIA','METAPHLAN4','GTDB','PHANTA-VIRUSES','KRAKEN2','ASSEMBLED-VIRUSES','GENBANK-VIRUSES','GENE-CATALOG',"KRAKEN2-CONF0.2-MASK","KRAKEN2-CONF0.0-MASK","KRAKEN2-CONF0.2-NOMASK","KRAKEN2-CONF0.0-NOMASK"))
summaryplotfodder$regclass2 = factor(summaryplotfodder$regclass2,levels = c('Overall','Oral','Nasal','Skin','Armpit','Belly Button','Forearm','Gluteal Crease','Nape of Neck','Post-Auricular','T-Zone','Toe Web Space'))

## SUPPLEMENT -- ALL RESULTS 

ggplot(summaryplotfodder,aes(y = timetrend, x = n, fill =seqtype)) + theme_bw() + geom_bar(stat='identity',position='dodge') + scale_x_log10()+ facet_grid(rows = vars(dset), cols = vars(regclass2)) + ylab('') + xlab('')+scale_fill_manual(values = c('darkblue','#BE1E2D'))
ggsave('~/Dropbox (Mason Lab)/i4/regressionsummary/all_counts_by_type_supp.pdf',width=14,height=16)

## TOTAL VIRAL SHIFTS BY CLASS
fortrendline_genvir = left_join(fortrendline %>% filter(dset == 'GENBANK-VIRUSES')  %>% mutate(genus = strsplit(yvar,';g__') %>% map_chr(2)) %>% mutate(phylum = strsplit(yvar,'p__') %>% map_chr(2) %>% strsplit(';') %>% map_chr(1)) ,annovir %>% select(-species,-phylum) %>% distinct,by = c('genus','phylum'='phylum2')) %>% mutate(regclass3 = if_else(regclass2 == 'Oral' | regclass2 == 'Nasal',regclass2,'Skin')) %>% filter(host == 'Prokaryotes' | host == 'Eukaryotes' )
summaryplotfodder_genvir_mol = fortrendline_genvir %>% select(dset,regclass2,timetrend,host) %>% group_by(dset,regclass2,timetrend,host) %>% count

#summaryplotfodder_genvir_mol$regclass3 = factor(summaryplotfodder_genvir_mol$regclass3,levels = c('Oral','Nasal','Skin'))

all_combinations <- expand_grid(timetrend = unique(summaryplotfodder_genvir_mol$timetrend),
                                regclass2 = unique(summaryplotfodder_genvir_mol$regclass2),
                                host = unique(summaryplotfodder_genvir_mol$host))

summaryplotfodder_genvir_mol <- left_join(all_combinations, summaryplotfodder_genvir_mol, by = c("timetrend", "regclass2", "host")) %>%replace_na(list(n = 0))

summaryplotfodder_genvir_mol$timetrend = gsub(' ','\n',summaryplotfodder_genvir_mol$timetrend)

summaryplotfodder_genvir_mol$regclass2 = factor(summaryplotfodder_genvir_mol$regclass2,levels = c('Oral','Nasal','Armpit','Belly Button','Forearm','Gluteal Crease','Nape of Neck','Post-Auricular','T-Zone','Toe Web Space'))

ggplot(summaryplotfodder_genvir_mol,aes(x = timetrend, y = n+1,fill = regclass2)) + theme_bw() + geom_bar(stat='identity',position='dodge')+ facet_grid(cols = vars(host)) + ylab('') + xlab('')+scale_fill_brewer(palette = 'Set3') + theme(axis.text.x = element_text(angle=60,hjust=1),legend.title = element_blank(),legend.position = 'none') + scale_y_log10()+ ylim(0,190)
ggsave('~/Dropbox (Mason Lab)/i4/regressionsummary/counts_by_viral_host.pdf',width=4,height=2)

fortrendline_genvir =left_join(fortrendline %>% filter(dset == 'GENBANK-VIRUSES')  %>% mutate(genus = strsplit(yvar,';g__') %>% map_chr(2)) %>% mutate(phylum = strsplit(yvar,'p__') %>% map_chr(2) %>% strsplit(';') %>% map_chr(1)) ,annovir %>% select(-species,-phylum) %>% distinct,by = c('genus','phylum'='phylum2')) %>% mutate(regclass3 = if_else(regclass2 == 'Oral' | regclass2 == 'Nasal',regclass2,'Skin')) %>% filter(molecule == 'DNA' | molecule == 'RNA' )
summaryplotfodder_genvir_mol = fortrendline_genvir %>% select(dset,regclass2,timetrend,molecule) %>% group_by(dset,regclass2,timetrend,molecule) %>% count

#summaryplotfodder_genvir_mol$regclass3 = factor(summaryplotfodder_genvir_mol$regclass3,levels = c('Oral','Nasal','Skin'))

all_combinations <- expand_grid(timetrend = unique(summaryplotfodder_genvir_mol$timetrend),
                                regclass2 = unique(summaryplotfodder_genvir_mol$regclass2),
                                molecule = unique(summaryplotfodder_genvir_mol$molecule))

summaryplotfodder_genvir_mol <- left_join(all_combinations, summaryplotfodder_genvir_mol, by = c("timetrend", "regclass2", "molecule")) %>%
  replace_na(list(n = 0))

summaryplotfodder_genvir_mol$timetrend = gsub(' ','\n',summaryplotfodder_genvir_mol$timetrend)

summaryplotfodder_genvir_mol$regclass2 = factor(summaryplotfodder_genvir_mol$regclass2,levels = c('Oral','Nasal','Armpit','Belly Button','Forearm','Gluteal Crease','Nape of Neck','Post-Auricular','T-Zone','Toe Web Space'))

ggplot(summaryplotfodder_genvir_mol,aes(x = timetrend, y = n+1,fill = regclass2)) + theme_bw() + geom_bar(stat='identity',position='dodge')+ facet_grid(cols = vars(molecule)) + ylab('') + xlab('')+scale_fill_brewer(palette = 'Set3') + theme(axis.text.x = element_text(angle=60,hjust=1),legend.title = element_blank(),legend.position = 'none')+ scale_y_log10() + ylim(0,225)
ggsave('~/Dropbox (Mason Lab)/i4/regressionsummary/counts_by_viral_mol.pdf',width=4,height=2)

## MAIN -- PRIMARY RESULTS AND TOTAL INCREASE

summaryplotfodder2 = summaryplotfodder %>% filter(dset == 'GTDB' | dset == 'GENBANK-VIRUSES' | dset == 'GENE-CATALOG', regclass2 != 'Skin' ,regclass2 != 'Overall')

summaryplotfodder2$dset = as.character(summaryplotfodder2$dset)

summaryplotfodder2$dset[summaryplotfodder2$dset == 'GTDB'] = 'Bacterial species'
summaryplotfodder2$dset[summaryplotfodder2$dset == 'GENBANK-VIRUSES'] = 'Viral genera'
summaryplotfodder2$dset[summaryplotfodder2$dset == 'GENE-CATALOG'] = 'Genes'

summaryplotfodder2$dset = factor(summaryplotfodder2$dset,levels = c('Bacterial species','Viral genera','Genes'))

all_combinations <- expand.grid(
  timetrend = unique(summaryplotfodder2$timetrend),
  seqtype = unique(summaryplotfodder2$seqtype),
  dset = unique(summaryplotfodder2$dset),
  regclass2 = unique(summaryplotfodder2$regclass2)
)

complete_data <- left_join(all_combinations, summaryplotfodder2, by=c("timetrend", "seqtype", "dset", "regclass2"))

ggplot(complete_data, aes(y = timetrend, x = n, fill = seqtype)) + theme_bw() + geom_bar(stat='identity', position=position_dodge(width=0.9), width=0.9) + scale_x_log10() + facet_grid(rows = vars(dset), cols = vars(regclass2)) + ylab('') + xlab('log10(Number of significant hits)') + scale_fill_manual(values = c('darkblue','#BE1E2D')) + theme(legend.position = 'None', axis.text.x = element_text(angle=60, hjust=1))
ggsave('~/Dropbox (Mason Lab)/i4/regressionsummary/sig_counts_by_type.pdf',width=12,height=4)

### find trajectory types
####### TIGHTEN UP P VALUE RESTRICTIONS TO MAKE SPLINES TRACTABLE

fortrendline2 =  fortrendline %>%filter(regclass2 != 'OVERALL') %>% mutate(mergeid = paste(yvar,dset,seqtype,regclass2)) 

gtdb_traj = simple_splines('GTDB',gtdb_metag,gtdb_metat,fortrendline2)
genvir_traj = simple_splines('GENBANK-VIRUSES',viral_metag,viral_metat,fortrendline2)
gene_traj = simple_splines('GENE-CATALOG',gene_metag,gene_metat,fortrendline2)

gtdb_genvir = bind_rows(gtdb_traj[[1]],genvir_traj[[1]]) #%>% filter(timetrend == 'Increased in flight' | timetrend == 'Decreased in flight')
#gtdb_genvir = gtdb_genvir %>% filter(yvar %in% (gtdb_genvir %>% filter(value!=0)%>% select(Crew.ID,yvar) %>% unique %>% group_by(yvar) %>% count %>% filter(n>1) %>% select(yvar) %>% unlist %>% unname))
temp = gene_traj[[1]] #%>% filter(timetrend == 'Increased in flight' | timetrend == 'Decreased in flight')
#temp = temp %>% filter(yvar %in% (temp %>% filter(value!=0)%>% select(Crew.ID,yvar) %>% unique %>% group_by(yvar) %>% csount %>% filter(n>1) %>% select(yvar) %>% unlist %>% unname))
temp1 =temp %>% group_by(timetrend,regclass2,seqtype) %>% slice_max(abs(est_pre),n=50)
temp2 =temp %>% group_by(timetrend,regclass2,seqtype) %>% slice_max(abs(est_post),n=50)
gene_traj[[1]] = bind_rows(temp1,temp2)
gtdb_genvir_gene = bind_rows(gene_traj[[1]],gtdb_genvir) %>% filter(regclass2!='Skin')

gtdb_genvir_gene$regclass2 = factor(gtdb_genvir_gene$regclass2,levels = c('Oral','Nasal','Armpit','Belly Button','Forearm','Gluteal Crease','Nape of Neck','Post-Auricular','T-Zone','Toe Web Space'))

## SIMPLIFY
gtdb_genvir_gene = gtdb_genvir_gene %>% mutate(regclass2=if_else(regclass2 == 'Oral' | regclass2 == 'Nasal' |regclass2 == 'Overall',regclass2,'Skin (all sites)'))

gtdb_genvir_gene$PLOTTYPE[gtdb_genvir_gene$PLOTTYPE == 'GENE-CATALOG'] = 'Genes'
gtdb_genvir_gene$PLOTTYPE[gtdb_genvir_gene$PLOTTYPE == 'GTDB'] = 'Bacterial\nspecies'
gtdb_genvir_gene$PLOTTYPE[gtdb_genvir_gene$PLOTTYPE == 'GENBANK-VIRUSES'] = 'Viral genera'

gtdb_genvir_gene$PLOTTYPE= factor(gtdb_genvir_gene$PLOTTYPE,levels = c('Bacterial\nspecies','Viral genera','Genes'))

gtdb_genvir_gene$regclass2 = factor(gtdb_genvir_gene$regclass2,levels = c('Oral','Nasal','Skin (all sites)'))

plot1 = ggplot(gtdb_genvir_gene %>% filter(timetrend == 'Persistent decrease' | timetrend == 'Persistent increase'),aes(x = Timepoint_Numeric,linetype = timetrend,color = seqtype,y=log(value+0.000001))) + geom_smooth(method = 'loess',alpha=.4,level=.95,position='jitter')  + facet_grid(seqtype + PLOTTYPE~regclass2 ,scales = 'free_y') + theme_bw() +scale_color_manual(values = c('darkblue','#BE1E2D')) + theme(legend.position = 'None')
ggsave(plot=plot1,'~/Dropbox (Mason Lab)/i4/regressionsummary/other_spline.pdf',width=4,height=5)

plot2 = ggplot(gtdb_genvir_gene%>% filter(timetrend != 'Persistent decrease' & timetrend != 'Persistent increase',timetrend !='NONE'),aes(x = Timepoint_Numeric,linetype = timetrend,color = seqtype,y=log(value+0.000001))) + geom_smooth(method = 'loess',alpha=.4,level=.95,position='jitter')  + facet_grid(seqtype + PLOTTYPE ~regclass2 ,scales = 'free_y') + theme_bw() +scale_color_manual(values = c('darkblue','#BE1E2D')) + theme(legend.position = 'None')
ggsave(plot=plot2,'~/Dropbox (Mason Lab)/i4/regressionsummary/inc_dec_spline.pdf',width=4,height=5)

```
######## UPSETR PLOTS
```{r}
### SUMMARY UPSET R PLOTS -- ALSO GENERATE BETWEEN DATA TYPES FOR SUPPLEMENT

# overall

check_argument = function(
    value,
    allowed,
    description
) {
    if (!(value %in% allowed)) {
        stop(
            paste0(
                description,
                ' has to be one of: ',
                paste(allowed, collapse=' or '),
                ', not "',
                value,
                '"'

            )
        )
    }
}


solve_mode = function (mode) {
  check_argument(
      mode,
      allowed = c(
          'exclusive_intersection', 'distinct',
          'inclusive_intersection', 'intersect',
          'exclusive_union', # no alias
          'inclusive_union', 'union'
      ),
      'mode'
  )

    # resolve aliases
    mode = switch(
        mode,
        distinct='exclusive_intersection',
        intersect='inclusive_intersection',
        union='inclusive_union',
        mode
    )
}


get_mode_presence = function(mode, prefix='in_', symbol=TRUE) {
    column = paste0(prefix, solve_mode(mode))
    if (symbol) {
        sym(column)
    } else {
        column
    }
}

presence = get_mode_presence('exclusive_intersection')

summarise_values = function(df) {
    aggregate(
        as.formula(paste0(presence, '~ intersection')),
        df,
        FUN=sum
    )
}

overall_intersections = regression_output %>% filter(BH_adjusted<0.05) %>% filter(regressionclass != 'Skin')  %>% filter(grepl('---',term)) %>% mutate(regclass2 = strsplit(as.character(term),' --- ') %>% map_chr(2)) %>% mutate(seqval =1 ) %>% dcast(yvar + direction+ launch  + dset + seqtype ~ regclass2,value.var ='seqval') 

overall_intersections = overall_intersections %>% mutate_if(is.numeric, ~1 * (. != 0))

overall_intersections[is.na(overall_intersections)]=0

overall_intersections2 = overall_intersections %>% filter(dset == 'GTDB') %>% distinct
pdf('~/Dropbox (Mason Lab)/i4/regressionsummary/gtdb_upset.pdf',width=5,height=4)
upset(overall_intersections2,set_sizes = FALSE,min_size=2,intersect = colnames(overall_intersections2)[6:ncol(overall_intersections2)],base_annotations=list('Intersection size'=intersection_size(counts=FALSE,mapping=aes(fill=seqtype))+theme(legend.position = 'None')+scale_fill_manual(values = c('darkblue','#BE1E2D'))),width_ratio=0.1)
dev.off()

overall_intersections3 = overall_intersections %>% filter(dset == 'GTDB',seqtype == 'METAGENOMICS') %>% distinct %>% mutate(idval = paste(yvar,direction,launch))
overall_intersections3[,c(16,6,7,8,9,10,11,12,13,14,15)] %>% column_to_rownames('idval') %>% rowSums%>% data.frame  %>% arrange(desc(.)) %>% head



overall_intersections2 = overall_intersections %>% filter(dset == 'GENBANK-VIRUSES') %>% distinct

pdf('~/Dropbox (Mason Lab)/i4/regressionsummary/genvir_upset.pdf',width=5,height=4)
upset(overall_intersections2,set_sizes = FALSE,min_size=2,intersect = colnames(overall_intersections2)[6:ncol(overall_intersections2)],base_annotations=list('Intersection size'=intersection_size(counts=FALSE,mapping=aes(fill=seqtype))+theme(legend.position = 'None')+scale_fill_manual(values = c('darkblue','#BE1E2D'))),width_ratio=0.1)
dev.off()

overall_intersections2 = overall_intersections %>% filter(dset == 'GENE-CATALOG') %>% distinct
### NOTE IN METHODS VARIATION IN INTERSECTION SIZE

pdf('~/Dropbox (Mason Lab)/i4/regressionsummary/genecat_upset.pdf',width=5,height=4)
upset(overall_intersections2,set_sizes = FALSE,min_size=50,intersect = colnames(overall_intersections2)[6:ncol(overall_intersections2)],base_annotations=list('Intersection size'=intersection_size(counts=FALSE,mapping=aes(fill=seqtype))+theme(legend.position = 'None')+scale_fill_manual(values = c('darkblue','#BE1E2D'))),width_ratio=0.1)
dev.off()

```

### FOLD CHANGE DOTPLOTS
```{r}
# high coverage dotplot regressions
categories = fortrendline %>% filter(regclass2 != 'OVERALL' &regclass2 != 'Overall',!grepl('LP',timetrendrank)) %>% select(yvar,seqtype,dset,timetrend,timetrendrank,regclass2,adj_pre,est_pre,adj_post,est_post)%>% filter(regclass2 != 'Skin') %>% mutate(regressionclass = if_else(regclass2 == 'Nasal' | regclass2 == 'Oral',regclass2,'Skin_Components'))

generate_fc_dotplot <- function(regression_output,dataset,metagdata,metatdata,regclass,categories,taxdata,countergrab){
    output=list()
  minval = 0.0000000001
  	temp_metat = regression_output %>% filter(BH_adjusted<0.05,!grepl('cds',yvar),!grepl('partial ORF',yvar),dset == dataset,seqtype == 'METATRANSCRIPTOMICS',regressionclass == regclass)%>% filter(grepl('---',term)) %>% mutate(regclass2 = strsplit(as.character(term),' --- ') %>% map_chr(2))
  	temp_metat = inner_join(temp_metat,categories %>% filter(dset == dataset,seqtype == 'METATRANSCRIPTOMICS',regressionclass == regclass))  %>% group_by(timetrend,regclass2) #%>% slice_min(BH_adjusted,n=50) %>%slice_max(abs(estimate),n=50)  %>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct
      	 	if(nrow(temp_metat)>0){
      	 	  print('Running metatranscriptomics')

    	temp2_metat_sub = metatdata %>% select(any_of(temp_metat$yvar)) %>% t %>% as.data.frame(check.names = F) %>% rownames_to_column('yvar') %>% melt %>% distinct
    	temp_metat = inner_join(temp_metat,temp2_metat_sub)
    	temp_metat = left_join(temp_metat,meta %>% rename(variable = SeqID)) %>% filter(!is.na(Timepoint))
    
    	# generate l2fc heatmap
    	l2fc_hm = temp_metat %>% filter(regclass2 == location)%>% group_by(yvar,Timepoint_Recode) %>% summarise(mean = mean(value)) %>% dcast(yvar ~ Timepoint_Recode,value.var ='mean') %>% mutate(l2fc_PRE = log2((`MID-FLIGHT`+minval)) - log2(`PRE-LAUNCH`+minval)) %>% mutate(l2fc_POST = -1*(log2((`MID-FLIGHT`+minval)) - log2(`POST-LAUNCH`+minval))) %>% select(yvar,l2fc_PRE,l2fc_POST) %>% mutate(diff = l2fc_POST - l2fc_PRE)%>% arrange(desc(abs((diff))))
    	l2fc_hm = left_join(l2fc_hm,taxdata,by=c('yvar'='query'))
    	l2fc_hm$Species = fct_reorder(as.factor(l2fc_hm$Species),l2fc_hm$diff)
    	#l2fc_hm$Genus = fct_reorder(as.factor(l2fc_hm$Genus),l2fc_hm$diff)
    	
   	temp =  left_join(l2fc_hm,temp_metat %>% select(yvar,timetrend) %>% distinct)  %>% mutate(seqtype = 'METATRANSCRIPTOMICS')  %>% mutate(ID  = paste(seqtype,yvar))%>% filter(timetrend == 'Transient decrease' & l2fc_PRE<0 & l2fc_POST>0 | timetrend == 'Transient increase' & l2fc_PRE>0 & l2fc_POST<0 | timetrend == 'Persistent increase' & l2fc_POST>0 |timetrend == 'Persistent decrease' & l2fc_PRE>0 |timetrend == 'Persistent increase' & l2fc_PRE<0|timetrend == 'Persistent increase' & l2fc_PRE<l2fc_POST |timetrend == 'Persistent decrease' & l2fc_PRE>l2fc_POST | timetrend == 'Persistent decrease' & l2fc_POST<0 ) %>% mutate(diff = if_else(l2fc_PRE<0 & l2fc_POST>0 & timetrend == "Persistent increase" |l2fc_PRE>0 & l2fc_POST<0 & timetrend == "Persistent decrease" ,diff + 10000000,diff)) %>% group_by(regclass2,timetrend)%>% distinct%>% slice_max(n=countergrab,abs(diff))
   	
   	# %>% filter(timetrend == 'Transient decrease' & l2fc_PRE<0 & l2fc_POST>0 | timetrend == 'Transient increase' & l2fc_PRE>0 & l2fc_POST<0 | timetrend == 'Persistent increase' & l2fc_POST>l2fc_PRE |timetrend == 'Persistent decrease' & l2fc_PRE>l2fc_POST) %>% group_by(regclass2,timetrend) 
   	
   	#%>% group_by(regclass2,timetrend) %>% 
     # temp2 = temp %>% dcast(yvar + timetrend +diff ~ variable,value.var = 'value' ) %>% filter(!(abs(l2fc_PRE) <1 & abs(l2fc_POST)<1)) 
     # temp2 = temp2 %>% filter(timetrend == 'Decreased in flight' & l2fc_PRE<0 & l2fc_POST>0 | timetrend == 'Increased in flight' & l2fc_PRE>0 & l2fc_POST<0 | timetrend == 'Increased in/after flight' & l2fc_POST>l2fc_PRE |timetrend == 'Decreased in/after flight' & l2fc_PRE>l2fc_POST)
    # TRY filtering based on log2fc (only show greater than abs 1 in one category, for the certain categories make sure log2fc is > 0 where it should be etc)
    	#temp = temp %>% filter(yvar %in% temp2$yvar)
    	output[['metat']] = temp
    	
    	}
  	temp_metag = regression_output %>% filter(BH_adjusted<0.05,dset == dataset,seqtype == 'METAGENOMICS',regressionclass == regclass)%>% filter(grepl('---',term)) %>% mutate(regclass2 = strsplit(as.character(term),' --- ') %>% map_chr(2))
  	temp_metag = inner_join(temp_metag,categories %>% filter(dset == dataset,seqtype == 'METAGENOMICS',regressionclass == regclass))  %>% group_by(timetrend,regclass2) #%>% slice_min(BH_adjusted,n=50) %>%slice_max(abs(estimate),n=50)  %>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct
      	 	if(nrow(temp_metag)>0){
      	 	  print('Running metagenomics')

    	temp2_metag_sub = metagdata %>% select(any_of(unique(temp_metag$yvar))) %>% t %>% as.data.frame(check.names = F) %>% rownames_to_column('yvar') %>% melt
    	temp_metag = inner_join(temp_metag,temp2_metag_sub)
    	temp_metag = left_join(temp_metag,meta %>% rename(variable = SeqID)) %>% filter(!is.na(Timepoint))
  
    	# generate l2fc heatmap
    	l2fc_hm = temp_metag %>% filter(regclass2 == location)%>% group_by(yvar,Timepoint_Recode) %>% summarise(mean = mean(value)) %>% dcast(yvar ~ Timepoint_Recode,value.var ='mean') %>% mutate(l2fc_PRE = log2((`MID-FLIGHT`+minval)) - log2(`PRE-LAUNCH`+minval)) %>% mutate(l2fc_POST = -1*(log2((`MID-FLIGHT`+minval)) - log2(`POST-LAUNCH`+minval))) %>% select(yvar,l2fc_PRE,l2fc_POST) %>% mutate(diff = l2fc_POST - l2fc_PRE)%>% arrange(desc(abs((diff))))
    	l2fc_hm = left_join(l2fc_hm,taxdata,by=c('yvar'='query'))
    	l2fc_hm$Species = fct_reorder(as.factor(l2fc_hm$Species),l2fc_hm$diff)
    	
   	temp =  left_join(l2fc_hm,temp_metag %>% select(yvar,timetrend) %>% distinct)  %>% mutate(seqtype = 'METAGENOMICS')  %>% mutate(ID  = paste(seqtype,yvar))%>% filter(timetrend == 'Transient decrease' & l2fc_PRE<0 & l2fc_POST>0 | timetrend == 'Transient increase' & l2fc_PRE>0 & l2fc_POST<0 | timetrend == 'Persistent increase' & l2fc_POST>0 |timetrend == 'Persistent decrease' & l2fc_PRE>0 |timetrend == 'Persistent increase' & l2fc_PRE<0|timetrend == 'Persistent increase' & l2fc_PRE<l2fc_POST |timetrend == 'Persistent decrease' & l2fc_PRE>l2fc_POST | timetrend == 'Persistent decrease' & l2fc_POST<0 ) %>% mutate(diff = if_else(l2fc_PRE<0 & l2fc_POST>0 & timetrend == "Persistent increase" |l2fc_PRE>0 & l2fc_POST<0 & timetrend == "Persistent decrease" ,diff + 10000000,diff)) %>% group_by(regclass2,timetrend)%>% distinct%>% slice_max(n=countergrab,abs(diff))
    	output[['metag']] = temp
      	 	}
  	return(output)
}

generate_fc_dotplot_v <- function(regression_output,dataset,metagdata,metatdata,regclass,categories,taxdata,countergrab){
    output=list()
  minval = 0.0000000001
  	temp_metat = regression_output %>% filter(BH_adjusted<0.05,!grepl('cds',yvar),!grepl('partial ORF',yvar),dset == dataset,seqtype == 'METATRANSCRIPTOMICS',regressionclass == regclass)%>% filter(grepl('---',term)) %>% mutate(regclass2 = strsplit(as.character(term),' --- ') %>% map_chr(2))
  	temp_metat = inner_join(temp_metat,categories %>% filter(dset == dataset,seqtype == 'METATRANSCRIPTOMICS',regressionclass == regclass))  %>% group_by(timetrend,regressionclass)#%>% slice_min(BH_adjusted,n=50) %>%slice_max(abs(estimate),n=50)  %>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct
      	 	if(nrow(temp_metat)>0){
      	 	  print('Running metatranscriptomics')

    	temp2_metat_sub = metatdata %>% select(any_of(temp_metat$yvar)) %>% t %>% as.data.frame(check.names = F) %>% rownames_to_column('yvar') %>% melt %>% distinct
    	temp_metat = inner_join(temp_metat,temp2_metat_sub)
    	temp_metat = left_join(temp_metat,meta %>% rename(variable = SeqID)) %>% filter(!is.na(Timepoint))
    
    	# generate l2fc heatmap
    	l2fc_hm = temp_metat %>% filter(regclass2 == location)%>% group_by(yvar,Timepoint_Recode) %>% summarise(mean = mean(value)) %>% dcast(yvar ~ Timepoint_Recode,value.var ='mean') %>% mutate(l2fc_PRE = log2((`MID-FLIGHT`+minval)) - log2(`PRE-LAUNCH`+minval)) %>% mutate(l2fc_POST = -1*(log2((`MID-FLIGHT`+minval)) - log2(`POST-LAUNCH`+minval))) %>% select(yvar,l2fc_PRE,l2fc_POST) %>% mutate(diff = l2fc_POST - l2fc_PRE)%>% arrange(desc(abs((diff))))
    	l2fc_hm = left_join(l2fc_hm,taxdata,by=c('yvar'='query'))
    	l2fc_hm$Species = fct_reorder(as.factor(l2fc_hm$Species),l2fc_hm$diff)
    	#l2fc_hm$Genus = fct_reorder(as.factor(l2fc_hm$Genus),l2fc_hm$diff)
    	
   	temp =  left_join(l2fc_hm,temp_metat %>% select(yvar,timetrend) %>% distinct)  %>% mutate(seqtype = 'METATRANSCRIPTOMICS')   %>% mutate(ID  = paste(seqtype,yvar))%>% filter(timetrend == 'Transient decrease' & l2fc_PRE<0 & l2fc_POST>0 | timetrend == 'Transient increase' & l2fc_PRE>0 & l2fc_POST<0 | timetrend == 'Persistent increase' & l2fc_POST>0 |timetrend == 'Persistent decrease' & l2fc_PRE>0 |timetrend == 'Persistent increase' & l2fc_PRE<0|timetrend == 'Persistent increase' & l2fc_PRE<l2fc_POST |timetrend == 'Persistent decrease' & l2fc_PRE>l2fc_POST | timetrend == 'Persistent decrease' & l2fc_POST<0 ) %>% mutate(diff = if_else(l2fc_PRE<0 & l2fc_POST>0 & timetrend == "Persistent increase" |l2fc_PRE>0 & l2fc_POST<0 & timetrend == "Persistent decrease" ,diff + 10000000,diff)) %>% group_by(regressionclass,timetrend)%>% distinct%>% slice_max(n=countergrab,abs(diff))
   	
   	# %>% filter(timetrend == 'Transient decrease' & l2fc_PRE<0 & l2fc_POST>0 | timetrend == 'Transient increase' & l2fc_PRE>0 & l2fc_POST<0 | timetrend == 'Persistent increase' & l2fc_POST>l2fc_PRE |timetrend == 'Persistent decrease' & l2fc_PRE>l2fc_POST) %>% group_by(regclass2,timetrend) 
   	
   	#%>% group_by(regclass2,timetrend) %>% 
     # temp2 = temp %>% dcast(yvar + timetrend +diff ~ variable,value.var = 'value' ) %>% filter(!(abs(l2fc_PRE) <1 & abs(l2fc_POST)<1)) 
     # temp2 = temp2 %>% filter(timetrend == 'Decreased in flight' & l2fc_PRE<0 & l2fc_POST>0 | timetrend == 'Increased in flight' & l2fc_PRE>0 & l2fc_POST<0 | timetrend == 'Increased in/after flight' & l2fc_POST>l2fc_PRE |timetrend == 'Decreased in/after flight' & l2fc_PRE>l2fc_POST)
    # TRY filtering based on log2fc (only show greater than abs 1 in one category, for the certain categories make sure log2fc is > 0 where it should be etc)
    	#temp = temp %>% filter(yvar %in% temp2$yvar)
    	output[['metat']] = temp
    	
    	}
  	temp_metag = regression_output %>% filter(BH_adjusted<0.05,dset == dataset,seqtype == 'METAGENOMICS',regressionclass == regclass)%>% filter(grepl('---',term)) %>% mutate(regclass2 = strsplit(as.character(term),' --- ') %>% map_chr(2))
  	temp_metag = inner_join(temp_metag,categories %>% filter(dset == dataset,seqtype == 'METAGENOMICS',regressionclass == regclass))  %>% group_by(timetrend,regressionclass) #%>% slice_min(BH_adjusted,n=50) %>%slice_max(abs(estimate),n=50)  %>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct
      	 	if(nrow(temp_metag)>0){
      	 	  print('Running metagenomics')

    	temp2_metag_sub = metagdata %>% select(any_of(unique(temp_metag$yvar))) %>% t %>% as.data.frame(check.names = F) %>% rownames_to_column('yvar') %>% melt
    	temp_metag = inner_join(temp_metag,temp2_metag_sub)
    	temp_metag = left_join(temp_metag,meta %>% rename(variable = SeqID)) %>% filter(!is.na(Timepoint))
  
    	# generate l2fc heatmap
    	l2fc_hm = temp_metag %>% filter(regclass2 == location)%>% group_by(yvar,Timepoint_Recode) %>% summarise(mean = mean(value)) %>% dcast(yvar ~ Timepoint_Recode,value.var ='mean') %>% mutate(l2fc_PRE = log2((`MID-FLIGHT`+minval)) - log2(`PRE-LAUNCH`+minval)) %>% mutate(l2fc_POST = -1*(log2((`MID-FLIGHT`+minval)) - log2(`POST-LAUNCH`+minval))) %>% select(yvar,l2fc_PRE,l2fc_POST) %>% mutate(diff = l2fc_POST - l2fc_PRE)%>% arrange(desc(abs((diff))))
    	l2fc_hm = left_join(l2fc_hm,taxdata,by=c('yvar'='query'))
    	l2fc_hm$Species = fct_reorder(as.factor(l2fc_hm$Species),l2fc_hm$diff)
    	
   	temp =  left_join(l2fc_hm,temp_metag %>% select(yvar,timetrend) %>% distinct)  %>% mutate(seqtype = 'METAGENOMICS') %>% mutate(ID  = paste(seqtype,yvar))%>% filter(timetrend == 'Transient decrease' & l2fc_PRE<0 & l2fc_POST>0 | timetrend == 'Transient increase' & l2fc_PRE>0 & l2fc_POST<0 | timetrend == 'Persistent increase' & l2fc_POST>0 |timetrend == 'Persistent decrease' & l2fc_PRE>0 |timetrend == 'Persistent increase' & l2fc_PRE<0|timetrend == 'Persistent increase' & l2fc_PRE<l2fc_POST |timetrend == 'Persistent decrease' & l2fc_PRE>l2fc_POST | timetrend == 'Persistent decrease' & l2fc_POST<0 ) %>% mutate(diff = if_else(l2fc_PRE<0 & l2fc_POST>0 & timetrend == "Persistent increase" |l2fc_PRE>0 & l2fc_POST<0 & timetrend == "Persistent decrease" ,diff + 10000000,diff)) %>% group_by(regressionclass,timetrend)%>% distinct%>% slice_max(n=countergrab,abs(diff)) %>%group_by(regressionclass,timetrend)%>% slice_max(n=countergrab,abs(diff))
    	output[['metag']] = temp
      	 	}
  	return(output)
}


### ORAL
gtdb_heatmap_oral = generate_fc_dotplot(regression_output,'GTDB',gtdb_metag %>% t %>% as.data.frame(check.names=F),gtdb_metat %>% t %>% as.data.frame(check.names=F),'Oral',categories %>% filter(!grepl('LP',timetrendrank)),gtdb_tax %>% distinct %>% select(query,Phylum,Species),15) %>% bind_rows %>% mutate(dset = 'ORAL')
#kraken_heatmap_oral = generate_fc_dotplot(regression_output,'KRAKEN2',kraken_metag,kraken_metat,'Oral',categories,kraken_tax %>% select(kraken_column_name,phylum) %>% rename(query = kraken_column_name, Phylum = phylum,Species = species))
genvir_heatmap_oral = generate_fc_dotplot(regression_output,'GENBANK-VIRUSES',viral_metag%>% t %>% as.data.frame(check.names=F),viral_metat %>% t %>% as.data.frame(check.names=F),'Oral',categories,genvir_tax %>% select(query,Phy,Gen,Spe) %>% distinct %>% rename(Phylum = Phy,Genus = Gen,Species = Spe),15)%>% bind_rows %>% mutate(dset = 'ORAL') %>% mutate(Species = strsplit(yvar,'g__') %>% map_chr(2))
metaph_heatmap_oral = generate_fc_dotplot(regression_output,'METAPHLAN4',metaph_metag %>% t %>% as.data.frame(check.names=F),metaph_metat %>% t %>% as.data.frame(check.names=F),'Oral',categories,metaph_tax %>% distinct %>% select(query,Phylum,Species),15) %>% bind_rows %>% mutate(dset = 'ORAL')


### NASAL
#gtdb_heatmap_nasal = generate_fc_dotplot(regression_output,'GTDB',gtdb_metag%>% t %>% as.data.frame(check.names=F),gtdb_metat%>% t %>% as.data.frame(check.names=F),'Nasal',categories,gtdb_tax %>% distinct %>% select(query,Phylum,Species),10)%>% mutate(dset = 'NASAL')
#kraken_heatmap_nasal = generate_fc_dotplot(regression_output,'KRAKEN2',kraken_metag,kraken_metat,'Nasal',categories,kraken_tax %>% select(kraken_column_name,phylum) %>% rename(query = kraken_column_name, Phylum = phylum,Species = species))
genvir_heatmap_nasal = generate_fc_dotplot(regression_output,'GENBANK-VIRUSES',viral_metag%>% t %>% as.data.frame(check.names=F),viral_metat %>% t %>% as.data.frame(check.names=F),'Nasal',categories,genvir_tax %>% select(query,Phy,Gen,Spe) %>% distinct %>% rename(Phylum = Phy,Genus = Gen,Species = Spe),15)%>% bind_rows %>% mutate(dset = 'NASAL') %>% mutate(Species = strsplit(yvar,'g__') %>% map_chr(2))
#metaph_heatmap_nasal = generate_fc_dotplot(regression_output,'METAPHLAN4',metaph_metag %>% t %>% as.data.frame(check.names=F),metaph_metat %>% t %>% as.data.frame(check.names=F),'Nasal',categories,metaph_tax %>% distinct %>% select(query,Phylum,Species)) %>% bind_rows %>% mutate(dset = 'NASAL')

### SKIN
gtdb_heatmap_skin_components = generate_fc_dotplot(regression_output,'GTDB',gtdb_metag%>% t %>% as.data.frame(check.names=F),gtdb_metat%>% t %>% as.data.frame(check.names=F),'Skin_Components',categories,gtdb_tax %>% distinct %>% select(query,Phylum,Species),10)%>% bind_rows%>% mutate(dset = 'SKIN')
#kraken_heatmap_skin_components = generate_fc_dotplot(regression_output,'KRAKEN2',kraken_metag,kraken_metat,'Skin_Components',categories,kraken_tax %>% select(kraken_column_name,phylum) %>% rename(query = kraken_column_name, Phylum = phylum,Species = species))
genvir_heatmap_skin_components = generate_fc_dotplot_v(regression_output,'GENBANK-VIRUSES',viral_metag%>% t %>% as.data.frame(check.names=F),viral_metat %>% t %>% as.data.frame(check.names=F),'Skin_Components',categories,genvir_tax %>% select(query,Phy,Gen,Spe) %>% distinct %>% rename(Phylum = Phy,Genus = Gen,Species = Spe),15)%>% bind_rows %>% mutate(dset = 'SKIN') %>% mutate(Species = strsplit(yvar,'g__') %>% map_chr(2))
metaph_heatmap_skin_components  = generate_fc_dotplot(regression_output,'METAPHLAN4',metaph_metag %>% t %>% as.data.frame(check.names=F),metaph_metat %>% t %>% as.data.frame(check.names=F),'Skin_Components',categories,metaph_tax %>% distinct %>% select(query,Phylum,Species),15) %>% bind_rows %>% mutate(dset = 'SKIN')


### BACTERIAL PLOT
#bacplot = bind_rows(gtdb_heatmap_oral,gtdb_heatmap_skin_components) %>% mutate(dset = factor(dset,levels = c('ORAL','NASAL','SKIN'))) %>% arrange(dset,timetrend,seqtype) %>% mutate(seqv = seq(1,nrow(.)))
bacplot = bind_rows(gtdb_heatmap_oral) %>% select(-Phylum,-ID)%>% mutate(dset = factor(dset,levels = c('ORAL','NASAL','SKIN'))) %>% arrange(dset,timetrend,seqtype) %>% ungroup %>% melt(id.vars = c('yvar','regclass2','timetrend','seqtype','dset','Species','diff')) 

bacplot$variable = gsub('l2fc_','',bacplot$variable)
bacplot$variable = factor(bacplot$variable,levels=c('PRE','POST'))

bacplot = bacplot %>% group_by(timetrend, seqtype) %>% mutate(Species = factor(Species, levels = unique(Species[order(diff)]))) %>% ungroup()
                                               
plot = ggplot(bacplot,aes(x = value, y = Species,shape = variable,fill = seqtype)) + geom_bar(stat='identity',position='dodge')  + theme_minimal() + xlab('Log2(Fold Change)') + geom_vline(xintercept = 0,linetype='dashed',color='gray') + theme(legend.position = 'None') + facet_grid(dset +timetrend~   variable,scales= 'free_y',space = 'free_y')+scale_color_manual(values = c('darkblue','#BE1E2D')) +scale_fill_manual(values = c('METAGENOMICS'= 'darkblue','METATRANSCRIPTOMICS'='#BE1E2D'))  + ylab('')

ggsave(plot = plot,'~/Dropbox (Mason Lab)/i4/regression_dotplots/bacplot_fc_oral.pdf',width=9,height=9)

### OTHER BACPLOTS

bacplot1 = bind_rows(gtdb_heatmap_skin_components) %>% select(-Phylum,-ID)%>% mutate(dset = factor(dset,levels = c('ORAL','NASAL','SKIN'))) %>% arrange(dset,timetrend,seqtype) %>% ungroup %>% melt(id.vars = c('yvar','timetrend','regclass2','seqtype','dset','Species','diff')) 

bacplot1$variable = gsub('l2fc_','',bacplot1$variable)
bacplot1$variable = factor(bacplot1$variable,levels=c('PRE','POST'))

bacplot1 = bacplot1 %>% group_by(timetrend, seqtype) %>% mutate(Species = factor(Species, levels = unique(Species[order(diff)]))) %>% ungroup()

plot = ggplot(bacplot1,aes(x = value, y = Species,shape = variable,fill = seqtype)) + geom_bar(stat='identity',position='dodge')  + theme_minimal() + xlab('Log2(Fold Change)') + geom_vline(xintercept = 0,linetype='dashed',color='gray') + theme(legend.position = 'None') + facet_grid(timetrend + regclass2  ~variable,scales= 'free_y',space = 'free_y')+scale_color_manual(values = c('darkblue','#BE1E2D')) +scale_fill_manual(values = c('METAGENOMICS'= 'darkblue','METATRANSCRIPTOMICS'='#BE1E2D'))  + ylab('')+ theme(strip.text.y = element_text(angle = 0))


ggsave(plot = plot,'~/Dropbox (Mason Lab)/i4/regression_dotplots/bacplot_fc_skin.pdf',width=8.5,height=11)

### VIRAL PLOT
viralplot = bind_rows(genvir_heatmap_oral,genvir_heatmap_nasal,genvir_heatmap_skin_components %>% rename(regclass2 = regressionclass)) %>% select(-Phylum,-ID,-Genus)%>% mutate(dset = factor(dset,levels = c('ORAL','NASAL','SKIN')))  %>% ungroup %>% melt(id.vars = c('yvar','timetrend','regclass2','seqtype','dset','Species','diff'))# %>% arrange(timetrend,regclass2,seqtype,Species,diff) %>% group_by(timetrend,regclass2,seqtype,Species) %>% arrange(diff) #%>% mutate(seqv = seq(1,nrow(.)))

viralplot$regclass2 = factor(viralplot$regclass2,levels = c('Oral','Nasal','Skin_Components'))
viralplot$variable = gsub('l2fc_','',viralplot$variable)
viralplot$variable = factor(viralplot$variable,levels=c('PRE','POST'))
viralplot$Species = gsub('Gen_of_','',viralplot$Species)

viralplot = viralplot %>% group_by(timetrend, regclass2, seqtype) %>% mutate(Species = factor(Species, levels = unique(Species[order(diff)]))) %>% ungroup()

#viralplot$Species = fct_reorder(viralplot$Species,viralplot$diff)

plot = ggplot(viralplot,aes(x = value, y =Species,shape = variable,fill = seqtype)) + geom_bar(stat='identity',position='dodge')  + theme_minimal() + xlab('Log2(Fold Change)') + geom_vline(xintercept = 0,linetype='dashed',color='gray') + theme(legend.position = 'None') + facet_grid(timetrend + regclass2 ~variable,scales= 'free_y',space = 'free_y')+scale_color_manual(values = c('darkblue','#BE1E2D')) +scale_fill_manual(values = c('METAGENOMICS'= 'darkblue','METATRANSCRIPTOMICS'='#BE1E2D'))  + ylab('')
ggsave(plot = plot,'~/Dropbox (Mason Lab)/i4/regression_dotplots/viral_fc.pdf',width=6,height=5)
#ggsave(plot = gtdb_heatmap_oral,'~/Dropbox (Mason Lab)/i4/regression_dotplots/gtdb_oral_dotplot_fc.pdf',width=7,height=15)

#ggsave(plot = gtdb_heatmap_nasal,'~/Dropbox (Mason Lab)/i4/regression_dotplots/gtdb_nasal_dotplot_fc.pdf',width=7,height=15)

#ggsave(plot = gtdb_heatmap_skin_components,'~/Dropbox (Mason Lab)/i4/regression_dotplots/gtdb_skin_dotplot_fc.pdf',width=7,height=15)

```
########FUNCTIONAL DOTPLOTS
```{r}
#regression_outputgene = regression_output_geneanno
#dataset = 'GENE-CATALOG'
#metagdata = gene_metag %>% t %>% as.data.frame(check.names=F)
#metatdata = gene_metat %>% t %>% as.data.frame(check.names=F)
#regclass = 'Skin_Components'
#taxdata =gene_annos %>% select(yvar,product) %>% distinct %>% filter(product != 'hypothetical protein') %>% rename(query=yvar,Species=product)

generate_fc_dotplot_gene <- function(regression_outputgene,dataset,metagdata,metatdata,regclass,categories,taxdata,countergrab){
  output=list()
  minval = 0.0000000001
  ######### METATRANSCRIPTOMICS
  	temp_metat = regression_outputgene %>% filter(BH_adjusted<0.05,!grepl('cds',yvar),!grepl('partial ORF',yvar),dset == dataset,seqtype == 'METATRANSCRIPTOMICS',regressionclass == regclass)
       if(regclass == 'Overall'){
        temp_metat = inner_join(temp_metat,categories %>% filter(dset == dataset,seqtype == 'METATRANSCRIPTOMICS',regressionclass == regclass))  %>% group_by(timetrend) %>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct
       }
       if(regclass != 'Overall'){
      temp_metat = inner_join(temp_metat,categories %>% filter(dset == dataset,seqtype == 'METATRANSCRIPTOMICS',regressionclass == regclass))  %>% group_by(timetrend,regclass2)%>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct#%>% slice_min(BH_adjusted,n=50) %>%slice_max(abs(estimate),n=15)  
      }      	 	
  	if(nrow(temp_metat)>0){
      	 	  print('Running metatranscriptomics')

    	temp2_metat_sub = metatdata %>% select(any_of(temp_metat$yvar)) %>% t %>% as.data.frame(check.names = F) %>% rownames_to_column('yvar') %>% melt
    	temp_metat = inner_join(temp_metat,temp2_metat_sub)
    	temp_metat = left_join(temp_metat,meta %>% rename(variable = SeqID)) %>% filter(!is.na(Timepoint))
    
    	# generate l2fc heatmap
       	l2fc_hm = temp_metat %>% filter(regclass2 == location)%>% group_by(yvar,Timepoint_Recode) %>% summarise(mean = mean(value)) %>% dcast(yvar ~ Timepoint_Recode,value.var ='mean') %>% mutate(l2fc_PRE = log2((`MID-FLIGHT`+minval)) - log2(`PRE-LAUNCH`+minval)) %>% mutate(l2fc_POST = -1*(log2((`MID-FLIGHT`+minval)) - log2(`POST-LAUNCH`+minval))) %>% select(yvar,l2fc_PRE,l2fc_POST) %>% mutate(diff = l2fc_POST - l2fc_PRE)%>% arrange(desc(abs((diff))))
    	l2fc_hm = left_join(l2fc_hm,taxdata,by=c('yvar'='query'))
    	l2fc_hm$Species = fct_reorder(as.factor(l2fc_hm$Species),l2fc_hm$diff)
    	
   	temp =  left_join(l2fc_hm,temp_metat %>% select(yvar,timetrend) %>% distinct)  %>% mutate(seqtype = 'METATRANSCRIPTOMICS')  %>% mutate(ID  = paste(seqtype,yvar))%>% filter(timetrend == 'Transient decrease' & l2fc_PRE<0 & l2fc_POST>0 | timetrend == 'Transient increase' & l2fc_PRE>0 & l2fc_POST<0 | timetrend == 'Persistent increase' & l2fc_POST>0 |timetrend == 'Persistent decrease' & l2fc_PRE>0 |timetrend == 'Persistent increase' & l2fc_PRE<0 | timetrend == 'Persistent decrease' & l2fc_POST<0 )  %>% group_by(regclass2,timetrend)%>% slice_max(n=countergrab,abs(diff))
    	output[['metat']] = temp
  	}
  	######## METAGENOMICS
       temp_metag = regression_outputgene %>% filter(BH_adjusted<0.05,!grepl('cds',yvar),!grepl('partial ORF',yvar),dset == dataset,seqtype == 'METAGENOMICS',regressionclass == regclass)#%>% filter(grepl('---',term)) %>% mutate(regclass2 = strsplit(as.character(term),' --- ') %>% map_chr(2))
       if(regclass == 'Overall'){
        temp_metag = inner_join(temp_metag,categories %>% filter(dset == dataset,seqtype == 'METAGENOMICS',regressionclass == regclass))  %>% group_by(timetrend) %>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct
       }
       if(regclass != 'Overall'){
      temp_metag = inner_join(temp_metag,categories %>% filter(dset == dataset,seqtype == 'METAGENOMICS',regressionclass == regclass))  %>% group_by(timetrend,regclass2)%>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct#%>% slice_min(BH_adjusted,n=50) %>%slice_max(abs(estimate),n=15)  %>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct
      }
       if(nrow(temp_metag)>0){
                    print('Running metagenomics')

      temp2_metag_sub = metagdata %>% select(any_of(temp_metag$yvar)) %>% t %>% as.data.frame(check.names = F) %>% rownames_to_column('yvar') %>% melt
      temp_metag = inner_join(temp_metag,temp2_metag_sub)
      temp_metag = left_join(temp_metag,meta %>% rename(variable = SeqID)) %>% filter(!is.na(Timepoint))
    
      # generate l2fc heatmap
       	l2fc_hm = temp_metag %>% filter(regclass2 == location)%>% group_by(yvar,Timepoint_Recode) %>% summarise(mean = mean(value)) %>% dcast(yvar ~ Timepoint_Recode,value.var ='mean') %>% mutate(l2fc_PRE = log2((`MID-FLIGHT`+minval)) - log2(`PRE-LAUNCH`+minval)) %>% mutate(l2fc_POST = -1*(log2((`MID-FLIGHT`+minval)) - log2(`POST-LAUNCH`+minval))) %>% select(yvar,l2fc_PRE,l2fc_POST) %>% mutate(diff = l2fc_POST - l2fc_PRE)%>% arrange(desc(abs((diff))))
    	l2fc_hm = left_join(l2fc_hm,taxdata,by=c('yvar'='query'))
    	l2fc_hm$Species = fct_reorder(as.factor(l2fc_hm$Species),l2fc_hm$diff)
    	
   	temp =  left_join(l2fc_hm,temp_metag %>% select(yvar,timetrend) %>% distinct)  %>% mutate(seqtype = 'METAGENOMICS')  %>% mutate(ID  = paste(seqtype,yvar))%>% filter(timetrend == 'Transient decrease' & l2fc_PRE<0 & l2fc_POST>0 | timetrend == 'Transient increase' & l2fc_PRE>0 & l2fc_POST<0 | timetrend == 'Persistent increase' & l2fc_POST>0 |timetrend == 'Persistent decrease' & l2fc_PRE>0 |timetrend == 'Persistent increase' & l2fc_PRE<0 | timetrend == 'Persistent decrease' & l2fc_POST<0 )  %>% group_by(regclass2,timetrend) %>% slice_max(n=countergrab,abs(diff))
    	output[['metag']] = temp
      	 	}
       return(output)
}

generate_fc_dotplot_gene_v <- function(regression_outputgene,dataset,metagdata,metatdata,regclass,categories,taxdata,countergrab){
  output=list()
  minval = 0.0000000001
  ######### METATRANSCRIPTOMICS
  	temp_metat = regression_outputgene %>% filter(BH_adjusted<0.05,!grepl('cds',yvar),!grepl('partial ORF',yvar),dset == dataset,seqtype == 'METATRANSCRIPTOMICS',regressionclass == regclass)
       if(regclass == 'Overall'){
        temp_metat = inner_join(temp_metat,categories %>% filter(dset == dataset,seqtype == 'METATRANSCRIPTOMICS',regressionclass == regclass))  %>% group_by(timetrend) %>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct
       }
       if(regclass != 'Overall'){
      temp_metat = inner_join(temp_metat,categories %>% filter(dset == dataset,seqtype == 'METATRANSCRIPTOMICS',regressionclass == regclass))  %>% group_by(timetrend,regressionclass)%>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct#%>% slice_min(BH_adjusted,n=50) %>%slice_max(abs(estimate),n=15)  
      }      	 	
  	if(nrow(temp_metat)>0){
      	 	  print('Running metatranscriptomics')

    	temp2_metat_sub = metatdata %>% select(any_of(temp_metat$yvar)) %>% t %>% as.data.frame(check.names = F) %>% rownames_to_column('yvar') %>% melt
    	temp_metat = inner_join(temp_metat,temp2_metat_sub)
    	temp_metat = left_join(temp_metat,meta %>% rename(variable = SeqID)) %>% filter(!is.na(Timepoint))
    
    	# generate l2fc heatmap
       	l2fc_hm = temp_metat %>% filter(regclass2 == location)%>% group_by(yvar,Timepoint_Recode) %>% summarise(mean = mean(value)) %>% dcast(yvar ~ Timepoint_Recode,value.var ='mean') %>% mutate(l2fc_PRE = log2((`MID-FLIGHT`+minval)) - log2(`PRE-LAUNCH`+minval)) %>% mutate(l2fc_POST = -1*(log2((`MID-FLIGHT`+minval)) - log2(`POST-LAUNCH`+minval))) %>% select(yvar,l2fc_PRE,l2fc_POST) %>% mutate(diff = l2fc_POST - l2fc_PRE)%>% arrange(desc(abs((diff))))
    	l2fc_hm = left_join(l2fc_hm,taxdata,by=c('yvar'='query'))
    	l2fc_hm$Species = fct_reorder(as.factor(l2fc_hm$Species),l2fc_hm$diff)
    	
   	temp =  left_join(l2fc_hm,temp_metat %>% select(yvar,timetrend) %>% distinct)  %>% mutate(seqtype = 'METATRANSCRIPTOMICS')  %>% mutate(ID  = paste(seqtype,yvar))%>% filter(timetrend == 'Transient decrease' & l2fc_PRE<0 & l2fc_POST>0 | timetrend == 'Transient increase' & l2fc_PRE>0 & l2fc_POST<0 | timetrend == 'Persistent increase' & l2fc_POST>0 |timetrend == 'Persistent decrease' & l2fc_PRE>0 |timetrend == 'Persistent increase' & l2fc_PRE<0 | timetrend == 'Persistent decrease' & l2fc_POST<0 )  %>% group_by(regressionclass,timetrend)%>% slice_max(n=countergrab,abs(diff))
    	output[['metat']] = temp
  	}
  	######## METAGENOMICS
       temp_metag = regression_outputgene %>% filter(BH_adjusted<0.05,!grepl('cds',yvar),!grepl('partial ORF',yvar),dset == dataset,seqtype == 'METAGENOMICS',regressionclass == regclass)#%>% filter(grepl('---',term)) %>% mutate(regclass2 = strsplit(as.character(term),' --- ') %>% map_chr(2))
       if(regclass == 'Overall'){
        temp_metag = inner_join(temp_metag,categories %>% filter(dset == dataset,seqtype == 'METAGENOMICS',regressionclass == regclass))  %>% group_by(timetrend) %>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct
       }
       if(regclass != 'Overall'){
      temp_metag = inner_join(temp_metag,categories %>% filter(dset == dataset,seqtype == 'METAGENOMICS',regressionclass == regclass))  %>% group_by(timetrend,regressionclass)%>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct#%>% slice_min(BH_adjusted,n=50) %>%slice_max(abs(estimate),n=15)  %>% select(timetrend,regressionclass,yvar,dset,regclass2,est_pre) %>% distinct
      }
       if(nrow(temp_metag)>0){
                    print('Running metagenomics')

      temp2_metag_sub = metagdata %>% select(any_of(temp_metag$yvar)) %>% t %>% as.data.frame(check.names = F) %>% rownames_to_column('yvar') %>% melt
      temp_metag = inner_join(temp_metag,temp2_metag_sub)
      temp_metag = left_join(temp_metag,meta %>% rename(variable = SeqID)) %>% filter(!is.na(Timepoint))
    
      # generate l2fc heatmap
       	l2fc_hm = temp_metag %>% filter(regclass2 == location)%>% group_by(yvar,Timepoint_Recode) %>% summarise(mean = mean(value)) %>% dcast(yvar ~ Timepoint_Recode,value.var ='mean') %>% mutate(l2fc_PRE = log2((`MID-FLIGHT`+minval)) - log2(`PRE-LAUNCH`+minval)) %>% mutate(l2fc_POST = -1*(log2((`MID-FLIGHT`+minval)) - log2(`POST-LAUNCH`+minval))) %>% select(yvar,l2fc_PRE,l2fc_POST) %>% mutate(diff = l2fc_POST - l2fc_PRE)%>% arrange(desc(abs((diff))))
    	l2fc_hm = left_join(l2fc_hm,taxdata,by=c('yvar'='query'))
    	l2fc_hm$Species = fct_reorder(as.factor(l2fc_hm$Species),l2fc_hm$diff)
    	
   	temp =  left_join(l2fc_hm,temp_metag %>% select(yvar,timetrend) %>% distinct)  %>% mutate(seqtype = 'METAGENOMICS')  %>% mutate(ID  = paste(seqtype,yvar))%>% filter(timetrend == 'Transient decrease' & l2fc_PRE<0 & l2fc_POST>0 | timetrend == 'Transient increase' & l2fc_PRE>0 & l2fc_POST<0 | timetrend == 'Persistent increase' & l2fc_POST>0 |timetrend == 'Persistent decrease' & l2fc_PRE>0 |timetrend == 'Persistent increase' & l2fc_PRE<0 | timetrend == 'Persistent decrease' & l2fc_POST<0 )  %>% group_by(regressionclass,timetrend) %>% slice_max(n=countergrab,abs(diff))
    	output[['metag']] = temp
      	 	}
       return(output)
}


regression_output_geneanno = inner_join(regression_output %>% filter(dset == 'GENE-CATALOG'),gene_annos %>% filter(product!='hypothetical protein') %>% distinct)
categories = fortrendline %>% filter(regclass2 != 'OVERALL',!grepl('LP',timetrendrank)) %>% select(yvar,seqtype,dset,timetrend,timetrendrank,regclass2,adj_pre,est_pre,adj_post,est_post)%>% filter(regclass2 != 'Skin') %>% mutate(regressionclass = if_else(regclass2 == 'Nasal' | regclass2 == 'Oral' | regclass2 == 'Overall',regclass2,'Skin_Components'))

#### QUICK SIDESTEP TO FIND KEGG GROUPS
kegg = regression_output_geneanno %>% filter(grepl('KEGG',other_annotations)) %>% select(yvar,other_annotations) %>% mutate(kegg = strsplit(other_annotations,'KEGG:') %>% map_chr(2) %>% strsplit(',') %>% map_chr(1)) %>% distinct
keggegories = inner_join(kegg,categories)

for(val in unique(keggegories$timetrend)){
  sub =keggegories %>% filter(timetrend == val)
  for(val2 in unique(keggegories$regressionclass)){
    sub2 =sub %>% filter(regressionclass == val2)
    for(val3 in unique(keggegories$seqtype)){
      sub3 =sub2 %>% filter(seqtype == val3)
    write.table(file = paste(gsub('/','',gsub(' ','',val)),val2,val3,'.tsv',sep=''),sub3 %>% select(kegg),quote=F,sep='\t',col.names = FALSE,row.names=FALSE) 
    }
  }
}

#gene_heatmap_overall = generate_fc_dotplot_gene(regression_output_geneanno,'GENE-CATALOG',gene_metag,gene_metat,'Overall',categories,gene_annos %>% select(yvar,product) %>% distinct %>% filter(product != 'hypothetical protein') %>% rename(query=yvar,Species=product)) %>% bind_rows %>% mutate(regclass = 'Overall')

# get prevalence for core genes
#prevs = gene_metag %>% select(all_of(gene_heatmap_overall %>% filter(seqtype == 'METAGENOMICS') %>% select(yvar) %>% unlist %>% unname %>% unique)) %>%  rownames_to_column('SeqID')%>% melt
#prevs = left_join(prevs,meta) %>% filter(!is.na(Crew.ID))
#prevs = prevs %>% mutate(skinoralnasal = if_else(isoral == 1,'ORAL',if_else(isskin==1,'SKIN','NASAL'))) 
#prevs = left_join(prevs %>% rename(yvar = variable), gene_heatmap_overall %>% select(yvar,Species))
#prevs = prevs %>% select(skinoralnasal,value,Timepoint_Recode,Species)%>% filter(Timepoint_Recode == 'MID-FLIGHT') %>% #mutate(value = if_else(value>0,1,0))
#prevs1 = prevs %>% group_by(skinoralnasal,Species) %>% summarize(prevalence = mean(value > 0), .groups = "drop")

#prevs = gene_metat %>% select(all_of(gene_heatmap_overall %>% filter(seqtype == 'METATRANSCRIPTOMICS') %>% select(yvar) %>% unlist %>% unname %>% unique)) %>%  rownames_to_column('SeqID')%>% melt
#prevs = left_join(prevs,meta) %>% filter(!is.na(Crew.ID))
#prevs = prevs %>% mutate(skinoralnasal = if_else(isoral == 1,'ORAL',if_else(isskin==1,'SKIN','NASAL'))) 
#prevs = left_join(prevs %>% rename(yvar = variable), gene_heatmap_overall %>% select(yvar,Species))
#prevs = prevs %>% select(skinoralnasal,value,Timepoint_Recode,Species)%>% filter(Timepoint_Recode == 'MID-FLIGHT') %>% #mutate(value = if_else(value>0,1,0))
#prevs2 = prevs %>% group_by(skinoralnasal,Species) %>% summarize(prevalence = mean(value > 0), .groups = "drop")

#prevs = bind_rows(prevs1,prevs2)

gene_heatmap_oral = generate_fc_dotplot_gene(regression_output_geneanno,'GENE-CATALOG',gene_metag %>% t %>% as.data.frame(check.names=F),gene_metat%>% t %>% as.data.frame(check.names=F),'Oral',categories,gene_annos %>% select(yvar,product) %>% distinct %>% filter(product != 'hypothetical protein') %>% rename(query=yvar,Species=product),10) %>% bind_rows %>% mutate(regclass = 'ORAL') 

gene_heatmap_nasal = generate_fc_dotplot_gene(regression_output_geneanno,'GENE-CATALOG',gene_metag %>% t %>% as.data.frame(check.names=F),gene_metat %>% t %>% as.data.frame(check.names=F),'Nasal',categories,gene_annos %>% select(yvar,product) %>% distinct %>% filter(product != 'hypothetical protein') %>% rename(query=yvar,Species=product),10) %>% bind_rows %>% mutate(regclass = 'NASAL')

gene_heatmap_skin = generate_fc_dotplot_gene_v(regression_output_geneanno,'GENE-CATALOG',gene_metag %>% t %>% as.data.frame(check.names=F),gene_metat %>% t %>% as.data.frame(check.names=F),'Skin_Components',categories,gene_annos %>% select(yvar,product) %>% distinct %>% filter(product != 'hypothetical protein') %>% rename(query=yvar,Species=product),10) %>% bind_rows %>% mutate(regclass = 'SKIN')

geneplot = bind_rows(gene_heatmap_oral) %>% select(-diff,-ID)%>% mutate(dset = factor(regclass,levels = c('ORAL','NASAL','SKIN'))) %>% arrange(dset,timetrend,seqtype) %>% select(-regclass) %>% ungroup %>% melt(id.vars = c('yvar','timetrend','regclass2','seqtype','dset','Species')) %>% mutate(seqv = seq(1,nrow(.)))

geneplot$Species = fct_reorder(geneplot$Species,geneplot$seqv)
geneplot$variable = gsub('l2fc_','',geneplot$variable)
geneplot$variable = factor(geneplot$variable,levels=c('PRE','POST'))

geneplot$regclass2 = factor(geneplot$regclass2,levels = c('Oral','Nasal','Skin'))

plot = ggplot(geneplot,aes(x = value, y = Species,shape = variable,fill = seqtype)) + geom_bar(stat='identity',position='dodge')  + theme_minimal() + xlab('Log2(Fold Change)') + geom_vline(xintercept = 0,linetype='dashed',color='gray') + theme(legend.position = 'None') + facet_grid(dset +timetrend~   variable,scales= 'free_y',space = 'free_y')+scale_color_manual(values = c('darkblue','#BE1E2D')) +scale_fill_manual(values = c('METAGENOMICS'= 'darkblue','METATRANSCRIPTOMICS'='#BE1E2D'))  + ylab('')
ggsave('~/Dropbox (Mason Lab)/i4/functional_plots/oral_functional.pdf',width=9,height=11)

geneplot = bind_rows(gene_heatmap_nasal) %>% select(-diff,-ID)%>% mutate(dset = factor(regclass,levels = c('ORAL','NASAL','SKIN'))) %>% arrange(dset,timetrend,seqtype) %>% select(-regclass) %>% ungroup %>% melt(id.vars = c('yvar','timetrend','regclass2','seqtype','dset','Species')) %>% mutate(seqv = seq(1,nrow(.)))

geneplot$Species = fct_reorder(geneplot$Species,geneplot$seqv)
geneplot$variable = gsub('l2fc_','',geneplot$variable)
geneplot$variable = factor(geneplot$variable,levels=c('PRE','POST'))

geneplot$regclass2 = factor(geneplot$regclass2,levels = c('Oral','Nasal','Skin'))

plot = ggplot(geneplot,aes(x = value, y = Species,shape = variable,fill = seqtype)) + geom_bar(stat='identity',position='dodge')  + theme_minimal() + xlab('Log2(Fold Change)') + geom_vline(xintercept = 0,linetype='dashed',color='gray') + theme(legend.position = 'None') + facet_grid(dset +timetrend~   variable,scales= 'free_y',space = 'free_y')+scale_color_manual(values = c('darkblue','#BE1E2D')) +scale_fill_manual(values = c('METAGENOMICS'= 'darkblue','METATRANSCRIPTOMICS'='#BE1E2D'))  + ylab('')
ggsave('~/Dropbox (Mason Lab)/i4/functional_plots/nasal_functional.pdf',width=9,height=11)

geneplot = bind_rows(gene_heatmap_skin) %>% select(-diff,-ID)%>% mutate(dset = factor(regclass,levels = c('ORAL','NASAL','SKIN'))) %>% arrange(dset,timetrend,seqtype) %>% select(-regclass) %>% ungroup %>% melt(id.vars = c('yvar','timetrend','regressionclass','seqtype','dset','Species')) %>% mutate(seqv = seq(1,nrow(.)))

geneplot$Species = fct_reorder(geneplot$Species,geneplot$seqv)
geneplot$variable = gsub('l2fc_','',geneplot$variable)
geneplot$variable = factor(geneplot$variable,levels=c('PRE','POST'))

#geneplot$regclass2 = factor(geneplot$regclass2,levels = c('Armpit','Belly Button','Forearm','Gluteal Crease','Nape of Neck','Post-Auricular','T-Zone','Toe Web Space'))

plot = ggplot(geneplot,aes(x = value, y = Species,shape = variable,fill = seqtype)) + geom_bar(stat='identity',position='dodge')  + theme_minimal() + xlab('Log2(Fold Change)') + geom_vline(xintercept = 0,linetype='dashed',color='gray') + theme(legend.position = 'None') + facet_grid(regressionclass +timetrend~   variable,scales= 'free_y',space = 'free_y')+scale_color_manual(values = c('darkblue','#BE1E2D')) +scale_fill_manual(values = c('METAGENOMICS'= 'darkblue','METATRANSCRIPTOMICS'='#BE1E2D'))  + ylab('')
ggsave('~/Dropbox (Mason Lab)/i4/functional_plots/skin_functional.pdf',width=9,height=11)


#facord1 = prevs %>% group_by(Species) %>% summarise(mean = mean(prevalence)) %>% arrange(desc(mean))
#facord2 = prevs%>% filter(prevalence>0) %>% select(skinoralnasal,Species) %>% distinct %>% group_by(Species) %>% count() %>% ungroup %>% select(Species,n)

#facord = inner_join(facord1,facord2) %>% arrange((n),(mean)) %>% mutate(seqs=seq(1,nrow(.)))
#facord$Species = fct_reorder(as.factor(facord$Species),facord$seqs)

#gene_heatmap_overall$Species = factor(gene_heatmap_overall$Species,levels = facord$Species)
#prevs$Species = factor(prevs$Species,levels = facord$Species)

#plot0 = ggplot(gene_heatmap_overall %>% filter(timetrend!='Decreased in flight'),aes(x = value, y = Species,shape = variable,fill = seqtype)) +geom_bar(stat='identity',position='dodge') + theme_bw() + xlab('Log2(Fold Change)') + geom_vline(xintercept = 0,linetype='dashed',color='gray') + theme(legend.position = 'None') + facet_grid( timetrend ~ .,scales= 'free') +scale_fill_manual(values = c('METAGENOMICS'= 'darkblue','METATRANSCRIPTOMICS'='#BE1E2D')) + ylab('') +facet_wrap(variable~.)

#plot00 =ggplot(prevs %>% mutate(variable= 'PRE')%>% filter(Species!='Ribosomal protein bL12, rplL'),aes(y = Species,x= prevalence,color=skinoralnasal)) + geom_point(alpha=.7,size=4) + theme_bw() +facet_wrap(variable~.) + scale_color_manual(values = c('#27ade3','#f5b183','#fce799'))
  
#plot_grid(plot0,plot00+ theme(axis.text.y = element_blank(),axis.ticks.y = element_blank(), axis.title.y = element_blank(),legend.position = 'None'), ncol=2, align='h',rel_widths = c(4,1))
#ggsave('~/Dropbox (Mason Lab)/i4/functional_plots/overall_functional.pdf',width=10,height=15)

```

```{r}

a = gene_annos %>% select(yvar,product) %>% distinct %>% filter(product!='hypothetical protein')
b= fortrendline%>% filter(regclass2 != 'Skin') %>% select(regclass2,timetrend,timetrendrank,yvar,seqtype) %>% distinct #%>% mutate(regressionclass = if_else(regclass2 == 'Nasal' | regclass2 == 'Oral' | regclass2 == 'Overall',regclass2,'Skin_Components'))

d = inner_join(a,b) %>% select(timetrend,regclass2,product,seqtype) 
full= inner_join(a,b)

increased_mg = d %>% filter(timetrend == 'Transient increase',seqtype == 'METAGENOMICS')%>% dcast(product ~ regclass2 ) %>% column_to_rownames('product')
increased_mg[increased_mg>=1]=1
decreased_mg = d %>% filter(timetrend == 'Transient decrease',seqtype == 'METAGENOMICS')%>% dcast(product ~ regclass2 ) %>% column_to_rownames('product')
decreased_mg[decreased_mg>=1]=1

increased_mt = d %>% filter(timetrend == 'Transient increase',seqtype == 'METATRANSCRIPTOMICS')%>% dcast(product ~ regclass2 ) %>% column_to_rownames('product')
increased_mt[increased_mt>=1]=1
decreased_mt = d %>% filter(timetrend == 'Transient decrease',seqtype == 'METATRANSCRIPTOMICS')%>% dcast(product ~ regclass2 ) %>% column_to_rownames('product')
decreased_mt[decreased_mt>=1]=1

q=increased_mg %>% filter(Oral == 1, Nasal == 1) %>% rowSums %>% data.frame %>% arrange(desc(.))# %>% filter(.>=6)  %>% mutate(timetrend = 'Transient increase',seqtype = 'METAGENOMICS')
w=decreased_mg %>% filter(Oral == 1, Nasal == 1) %>% rowSums %>% data.frame %>% arrange(desc(.)) #%>% filter(.>=6) %>% mutate(timetrend = 'Transient decrease',seqtype = 'METAGENOMICS')

e=increased_mt %>% filter(Oral == 1, Nasal == 1) %>% rowSums %>% data.frame %>% arrange(desc(.)) #%>% filter(.>=6)   %>% mutate(timetrend = 'Transient increase',seqtype = 'METATRANSCRIPTOMICS')
r=decreased_mt %>% filter(Oral == 1, Nasal == 1) %>% rowSums %>% data.frame %>% arrange(desc(.)) #%>% filter(.>=6) %>% mutate(timetrend = 'Transient decrease',seqtype = 'METATRANSCRIPTOMICS')

t = bind_rows(q,w,e,r)
write.csv(t,'~/Dropbox (Mason Lab)/i4/revisions/core_gene_flight.csv')

toxinantitoxin = full %>% filter(grepl('toxin',product),grepl('antitoxin',product),grepl('system',product,regclass2),!grepl('LP',timetrendrank)) %>% select(timetrend,seqtype,regclass2) %>% table %>% data.frame 
toxinantitoxin = left_join(toxinantitoxin,full %>% filter(grepl('toxin',product),grepl('antitoxin',product),grepl('system',product,regclass2),!grepl('LP',timetrendrank))%>% select(product,seqtype) %>% distinct %>% group_by(seqtype) %>% count %>% dplyr::rename(seqtypetotal = n))%>% mutate(class = paste0('Toxin-Antitoxin',' N = (',seqtypetotal,')'))

heme = full %>% filter(grepl('heme',product),!grepl('LP',timetrendrank)) %>% select(timetrend,seqtype,regclass2) %>% table %>% data.frame 
heme = left_join(heme,full %>% filter(grepl('heme',product),!grepl('LP',timetrendrank)) %>% select(product,seqtype) %>% distinct %>% group_by(seqtype) %>% count %>% dplyr::rename(seqtypetotal = n)) %>% mutate(class = paste0('Heme',' N = (',seqtypetotal,')'))

phage = full %>% filter(grepl('phage',product)|grepl('Phage',product),!grepl('LP',timetrendrank)) %>% select(timetrend,seqtype,regclass2) %>% table %>% data.frame 
phage = left_join(phage,full%>% filter(grepl('phage',product)|grepl('Phage',product),!grepl('LP',timetrendrank))%>% select(product,seqtype) %>% distinct %>% group_by(seqtype) %>% count %>% dplyr::rename(seqtypetotal = n)) %>% mutate(class = paste0('Phage',' N = (',seqtypetotal,')'))

lantibiotic = full %>% filter(grepl('lantibiotic',product)| grepl('Lantibiotic',product),!grepl('LP',timetrendrank))%>% select(timetrend,seqtype,regclass2) %>% table %>% data.frame
lantibiotic = left_join(lantibiotic, full%>% filter(grepl('lantibiotic',product)| grepl('Lantibiotic',product),!grepl('LP',timetrendrank))%>% select(product,seqtype) %>% distinct %>% group_by(seqtype) %>% count %>% dplyr::rename(seqtypetotal = n)) %>% mutate(class = paste0('Lantibiotic',' N = (',seqtypetotal,')'))

drug_metal_resistnace = full %>% filter(grepl('resistance',product)| grepl('Resistance',product),!grepl('LP',timetrendrank))%>% select(timetrend,seqtype,regclass2) %>% table %>% data.frame
drug_metal_resistnace = left_join(drug_metal_resistnace, full %>% filter(grepl('resistance',product)| grepl('Resistance',product),!grepl('LP',timetrendrank))%>% select(product,seqtype) %>% distinct %>% group_by(seqtype) %>% count %>% dplyr::rename(seqtypetotal = n)) %>% mutate(class = paste0('Antibiotic and heavy\nmetal resistance',' N = (',seqtypetotal,')'))

full2 = bind_rows(toxinantitoxin,heme,phage,lantibiotic,drug_metal_resistnace)# %>% mutate(newclass = paste0(class,' (N = ',total,')'))
full2$timetrend = gsub(' ','\n',full2$timetrend)

ggplot(full2,aes(x = timetrend, y = Freq, fill = regclass2)) + geom_bar(stat='identity',position = 'stack') + facet_wrap(seqtype ~ class,scales = 'free_y',nrow=2) + theme_minimal()+ theme(axis.text.x = element_text(angle = 60,hjust = 1),legend.position = 'bottom') + scale_fill_brewer(palette = 'Set3')
ggsave('~/Dropbox (Mason Lab)/i4/revisions/gene_grouping_plot.pdf',width=12,height=7)
```

######### COGS
```{r}
cogs = gene_annos %>% filter(grepl('COG',other_annotations))
d <- cogs %>% separate_rows(other_annotations, sep = ", ") %>% mutate(COG = ifelse(str_detect(other_annotations, "^COG:COG[0-9]+"), str_replace(str_extract(other_annotations, "^COG:COG[0-9]+"), "COG:", ""), NA_character_), COG_CLASS = ifelse(str_detect(other_annotations, "^COG:[A-Z]+"), str_replace(str_extract(other_annotations, "^COG:[A-Z]+$"), "COG:", ""), NA_character_)) %>% group_by(yvar, gene_name, product) %>% summarise(COG = first(na.omit(COG)), COG_CLASS = first(na.omit(COG_CLASS))) %>% ungroup()
d2 = d %>% select(yvar,COG_CLASS)
d2 = inner_join(d2,fortrendline %>% select(yvar,seqtype,regclass2,timetrend)) %>% filter(!is.na(timetrend))
d2 = d2 %>% mutate(splitcogs = strsplit(as.character(COG_CLASS), "")) %>% unnest(splitcogs)

cogmap=read.csv('~/Dropbox (Mason Lab)/i4/i4_data_packet/gene_catalog/cog_map.csv',header=F) 
colnames(cogmap) = c('GROUP','splitcogs','CATEGORY')
forimmunelater = left_join(d2,cogmap)%>% ungroup%>% filter(!is.na(CATEGORY)) %>% filter(regclass2!= 'Skin')
  
d2 =d2 %>% select(timetrend,seqtype,regclass2,splitcogs) %>% group_by(timetrend,seqtype,regclass2,splitcogs) %>% count %>% filter(regclass2 != 'Overall') %>% arrange(n)


d3 = left_join(d2,cogmap)%>% ungroup%>% filter(!is.na(CATEGORY)) %>% filter(regclass2!= 'Skin')
d3 = d3 %>% filter(!is.na(GROUP),!grepl('/',timetrend))

#foo$CATEGORY = fct_reorder(as.factor(foo$CATEGORY),foo$n)
#d3$CATEGORY = factor(d3$CATEGORY,levels = levels(foo$CATEGORY))

d3$regclass2 = factor(d3$regclass2,levels = c('Oral','Nasal','Armpit','Belly Button','Forearm','Gluteal Crease','Nape of Neck','Post-Auricular','T-Zone','Toe Web Space'))

ggplot(data = d3 %>% filter(grepl('Transient',timetrend)),aes(x = n,y = CATEGORY,fill=regclass2)) + geom_bar(stat='identity',width=.5) + theme_minimal() + facet_grid(GROUP ~ timetrend+seqtype ,scales='free',space = 'free_y') + scale_fill_manual(values = brewer.pal('Set3',n=11)) + xlab('') + theme(legend.position ='none')
ggsave("~/Dropbox (Mason Lab)/i4/revisions/summary_cog_cats.pdf",width=10,height=4)
```
####### IMMUNE STUFF
```{r} 
# immune cell interaction

parse_output <- function(dfile,seqtyped){
  immunecell = gsub('PI_','',dfile)
  immunecell = gsub('.gene.expression.orig.ident.csv','',immunecell)
  immunecell = gsub('DEC_','',immunecell)
  immunecell = gsub('INC_','',immunecell)
  immunecell = gsub('LASSO_OUT','',immunecell)
  immunecell = gsub('VIRUSES','',immunecell)
  immunecell = gsub('VIRAL','',immunecell)
  immunecell = gsub('METATRANSCRIPTOMIC','',immunecell)
  immunecell = gsub('METAGENOMIC','',immunecell)
  immunecell = gsub('average','',immunecell)
  immunecell = gsub('_','',immunecell)
  immunecell = gsub('\\.','',immunecell)
  dir = 'INCREASED'
  if(grepl('DEC',dfile)){
    dir='DECREASED'
  }
  data = read.csv(dfile) %>% mutate(seqtype = seqtyped)%>% filter(features!='(Intercept)') %>% mutate(directiontoflight = dir) %>% mutate(celltype =immunecell) %>% mutate(dset = 'GENE-CATALOG',dset = if_else(grepl('d__Bacteria',microbialfeaturename),'BACTERIAL -- GTDB',dset)) %>% mutate(dset = if_else(grepl('k__Bacteria',microbialfeaturename),'Bacterial -- METAPHLAN4',dset)) %>% mutate(dset = if_else(grepl('p__',microbialfeaturename) & dset == 'GENE-CATALOG','VIRAL',dset))
  if(nrow(data)==0){
    return('none')
  }
  x = data
  # count number of activated host genes 

  counts = data %>% select(regressionclass) %>% table %>% data.frame %>% mutate(seqtype = seqtyped)%>% mutate(celltype =immunecell)
  colnames(counts)[1] = 'Site Type'
  colnames(counts)[2] = 'Number of associated genes'
  
  # count features with most genes
  #c#ounts2 =data %>% select(regressionclass,microbialfeaturename) %>% table %>% data.frame %>% filter(Freq>1) %>% group_by(regressionclass) %>% slice_max(n=3,Freq) %>% mutate(seqtype = seqtyped)%>% mutate(celltype =immunecell)
  counts2 =data %>% select(regressionclass,microbialfeaturename) %>% table %>% data.frame  %>% group_by(regressionclass) %>% mutate(seqtype = seqtyped)%>% mutate(celltype =immunecell)
  colnames(counts2)[3] = 'Number of associated genes'
  
  # report most strongly activated host genes
  #maxes = data %>% group_by(regressionclass) %>% slice_max(n=3,coefficient) %>% mutate(seqtype = seqtyped)%>% mutate(celltype =immunecell)
  #mins = data %>% group_by(regressionclass)  %>% slice_min(n=3,coefficient) %>% mutate(seqtype = seqtyped)%>% mutate(celltype =immunecell)
  
  maxes = data %>% group_by(regressionclass) %>% mutate(seqtype = seqtyped)%>% mutate(celltype =immunecell)
  mins = data %>% group_by(regressionclass) %>% mutate(seqtype = seqtyped)%>% mutate(celltype =immunecell)
  
  
  data = data  %>% dcast(microbialfeaturename + regressionclass + seqtype~ features, value.var = 'coefficient')
  data[is.na(data)] = 0
  return(list(counts,counts2,bind_rows(maxes,mins),x))
}

setwd('~/Dropbox (Mason Lab)/i4/revisions/immune_output//')

metagparse = list()
files = list.files()
files = files[grepl('LASSO',files)]
files = files[grepl('METAGENOMIC',files)]
for(f in files){
  out = parse_output(f,'METAGENOMICS')
  if(typeof(out)!='character'){
      metagparse[[f]] = out
  }
}
 
metatparse = list()
files = list.files()
files = files[grepl('LASSO',files)]
files = files[grepl('METATRANSCRIPTOMIC',files)]
for(f in files){
  out = parse_output(f,'METATRANSCRIPTOMICS')
  if(typeof(out)!='character'){
      metatparse[[f]] = out
  }
}

metagout = bind_rows(map(metagparse, function(x) x[[4]])) %>% select(-any_of(c("X","X.1")))
metatout = bind_rows(map(metatparse, function(x) x[[4]]))%>% select(-any_of(c("X","X.1")))

alldata = bind_rows(metagout,metatout)

foo = expand.grid(unique(alldata$seqtype),unique(alldata$regressionclass),unique(alldata$dset),unique(alldata$directiontoflight),unique(alldata$celltype))
colnames(foo) =c('seqtype','regressionclass','dset','directiontoflight','celltype')

alldata = left_join(foo,alldata)
alldata[is.na(alldata)]=0

### NEW BUG FIX?
alldata = alldata %>% filter(microbialfeaturename!=0) %>% filter(celltype!='pbmc')

alldata$dset = factor(alldata$dset,levels = c('BACTERIAL -- GTDB','Bacterial -- METAPHLAN4','VIRAL','GENE-CATALOG'))
alldata$regressionclass = factor(alldata$regressionclass,levels = c('Oral','Nasal','Skin'))
alldata$celltype = gsub('T',' T',alldata$celltype)
alldata$celltype = gsub('Mono',' Mono',alldata$celltype)
alldata$celltype = factor(alldata$celltype,levels = c("CD4 T", "CD8 T", "other T", "B", "NK", "CD14 Mono", "CD16 Mono", "DC", "other","pbmc"))

pibugs = fortrendline %>% filter(!grepl('LP',timetrendrank)) %>% filter(timetrend == 'Persistent increase' | timetrend == 'Transient increase') %>% mutate(regressionclass = if_else(regclass2 != 'Oral' & regclass2 != 'Capsule' & regclass2 != 'Nasal','Skin',regclass2)) %>% select(regressionclass,yvar,seqtype) %>% rename(microbialfeaturename = yvar) %>% distinct 

#alldata = alldata %>% filter(microbialfeaturename %in% pibugs$yvar)
alldata = inner_join(alldata,pibugs)
alldata$regressionclass = factor(alldata$regressionclass,levels = c('Oral','Nasal','Skin'))

#### INCREASED FEATURES
celltypehitscount = alldata %>% filter(directiontoflight == 'INCREASED') %>% group_by(seqtype,dset,regressionclass,celltype) %>% count() %>% ungroup %>% dcast(seqtype +dset +regressionclass ~celltype,value.var = "n")
celltypehitscount[is.na(celltypehitscount)] = 0
celltypehitscount = melt(celltypehitscount,id.vars = c("seqtype","dset","regressionclass")) %>% rename(celltype = variable)


all_combinations <- expand_grid(regressionclass = unique(celltypehitscount$regressionclass),celltype = unique(celltypehitscount$celltype),seqtype = unique(celltypehitscount$seqtype),dset = unique(celltypehitscount$dset))

fullcounts_expanded <- left_join(all_combinations, celltypehitscount, by = c("regressionclass", "seqtype", "dset", "celltype")) %>% replace_na(list(n = 0))

ggplot(fullcounts_expanded,aes(y=value+1,x=dset,fill=seqtype)) + theme_minimal() + geom_bar(stat='identity',position = position_dodge(preserve = "single")) + facet_grid(dset ~ celltype,scales='free_y',space='free_x') + theme(legend.position = 'bottom',axis.text.x = element_text(angle = 60,hjust = 1))+scale_fill_manual(values = c('METAGENOMICS'= 'darkblue','METATRANSCRIPTOMICS'='#BE1E2D')) + ggtitle('Immune cell gene expression and microbial features increased after flight') + ylab('# associated genes')


#ggplot(fullcounts_expanded,aes(y=value+1,x=regressionclass,fill=seqtype)) + theme_minimal() + geom_bar(stat='identity',position = position_dodge(preserve = "single")) + facet_grid(dset ~ celltype,scales='free_y',space='free_x') + theme(legend.position = 'bottom',axis.text.x = element_text(angle = 60,hjust = 1))+scale_fill_brewer(palette = 'Set1') + ggtitle('Immune cell gene expression and microbial features increased after flight')

ggsave('~/Dropbox (Mason Lab)/i4/immune_modeling/top_counts_increase.pdf',width=6,height=4)

### GET MOST IMPACTFUL MICROBIAL EXPRESSION CHANGES
foo = left_join(alldata,gene_annos,c('microbialfeaturename' ='yvar'))
impactfulbugs = foo %>% filter(features!="0")%>% filter(dset == 'VIRAL' | dset == 'BACTERIAL -- GTDB' | dset == 'GENE-CATALOG'| product != 'hypothetical protein')%>% group_by(seqtype,dset,regressionclass,microbialfeaturename) %>% count() %>% group_by(seqtype,dset,regressionclass) %>% arrange(n)%>% ungroup %>% select(-n)

#temp = impactfulbugs %>% filter(dset == 'GENE-CATALOG')  %>% select(microbialfeaturename)%>% unlist %>% unname
#temp2 = impactfulbugs %>% filter(dset == 'VIRAL') %>% mutate(microbialfeaturename = strsplit(microbialfeaturename,',') %>% map_chr(1)) %>% select(microbialfeaturename)%>% unlist %>% unname
#temp3 = impactfulbugs %>% filter(dset == 'BACTERIAL') %>% mutate(microbialfeaturename = strsplit(microbialfeaturename,'s__') %>% map_chr(2)) %>% select(microbialfeaturename)%>% unlist %>% unname

#ordering = c(temp,temp2,temp3)

impactfulbugplot = left_join(impactfulbugs,alldata %>% select(seqtype,dset,regressionclass,microbialfeaturename,celltype,features,directiontoflight) %>% filter(!is.na(features))) %>% distinct
impactfulbugplot = impactfulbugplot %>% group_by(seqtype,dset,regressionclass,microbialfeaturename,celltype) %>% count()

temp = impactfulbugplot %>% filter(dset == 'GENE-CATALOG') 
temp2 = impactfulbugplot %>% filter(dset == 'VIRAL') %>% mutate(microbialfeaturename = strsplit(microbialfeaturename,',') %>% map_chr(1))
temp3 = impactfulbugplot %>% filter(dset == 'BACTERIAL -- GTDB') %>% mutate(microbialfeaturename = strsplit(microbialfeaturename,'s__') %>% map_chr(2))
#temp4 = impactfulbugplot %>% filter(dset == 'Bacterial -- METAPHLAN4') %>% mutate(microbialfeaturename = strsplit(microbialfeaturename,'s__') %>% map_chr(2))

impactfulbugplot = bind_rows(temp,temp2,temp3) 

impactfulbugplot$microbialfeaturename = factor(impactfulbugplot$microbialfeaturename,levels = unique(impactfulbugplot$microbialfeaturename))
impactfulbugplot$dset = factor(impactfulbugplot$dset,levels = c('BACTERIAL -- GTDB','VIRAL','GENE-CATALOG'))
impactfulbugplot$regressionclass = factor(impactfulbugplot$regressionclass,levels = c('Oral','Nasal','Skin'))
#impactfulbugplot$celltype = gsub('T',' T',impactfulbugplot$celltype)
#impactfulbugplot$celltype = gsub('Mono',' Mono',impactfulbugplot$celltype)
#impactfulbugplot$celltype = factor(impactfulbugplot$celltype,levels = c("CD4 T", "CD8 T", "other T", "B", "NK", "CD14 Mono", "CD16 Mono", "DC", "other"))

impactfulbugplot = left_join(impactfulbugplot,gene_annos,c('microbialfeaturename' ='yvar')) %>% filter(product != 'hypothetical protein' | is.na(product))
impactfulbugplot = impactfulbugplot %>% mutate(microbialfeaturename = if_else(!is.na(product),product,as.character(microbialfeaturename)))

#get plot factor order, again I guess
foobar = impactfulbugplot %>% group_by(seqtype,dset,regressionclass,microbialfeaturename) %>% summarise(n=sum(n))
foobar$microbialfeaturename = fct_reorder(as.factor(foobar$microbialfeaturename),foobar$n)

impactfulbugplot$microbialfeaturename = factor(impactfulbugplot$microbialfeaturename ,levels = levels(foobar$microbialfeaturename))

impactfulbugplotsub = impactfulbugplot %>% filter(dset == 'VIRAL') %>% mutate(microbialfeaturename = microbialfeaturename%>% as.character %>% strsplit('g__')%>% map_chr(2)) 
impactfulbugplotsub$microbialfeaturename = gsub('Gen_unclassified','',impactfulbugplotsub$microbialfeaturename)
impactfulbugplotsub$microbialfeaturename = gsub('Gen_of_','',impactfulbugplotsub$microbialfeaturename)
impactfulbugplotsub2 = impactfulbugplot %>% filter(dset != 'VIRAL') 

impactfulbugplot = bind_rows(impactfulbugplotsub,impactfulbugplotsub2)  %>% filter(dset != 'Bacterial -- METAPHLAN4')

#### get cell type by seq type
#ggplot(impactfulbugplot %>% filter(dset!='VIRAL') %>% group_by(seqtype,celltype,dset) %>% count,aes(x = celltype,fill=seqtype,y=n)) + theme_minimal()+ geom_bar(stat='identity',position='dodge') +scale_fill_manual(values = c('darkblue','#BE1E2D'))+ facet_wrap(. ~ dset,ncol=1) + theme(axis.text.x = element_text(angle=60,hjust=1))

tokeep = impactfulbugplot %>% ungroup %>% group_by(regressionclass,seqtype,dset,microbialfeaturename) %>% summarise(tot = sum(n)) %>% slice_max(tot,n=10) %>% ungroup %>% select(microbialfeaturename,regressionclass) %>% distinct

impactfulbugplot = inner_join(impactfulbugplot,tokeep)

bing = impactfulbugplot %>% ungroup %>% group_by(dset,seqtype,microbialfeaturename) %>% summarise(tot = sum(n))  %>% arrange((tot))%>% arrange(dset,seqtype)

impactfulbugplot$microbialfeaturename = factor(impactfulbugplot$microbialfeaturename,levels = unique(bing$microbialfeaturename))


ggplot(impactfulbugplot %>% filter(dset != 'GENE-CATALOG'),aes(x = n,y =microbialfeaturename,fill = celltype))+ geom_bar(stat='identity') + facet_grid(dset +regressionclass~ seqtype  ,scales = 'free_y',space = 'free_y',switch = 'y') + theme_minimal() + scale_fill_brewer(palette = 'Set3') + xlab('Count') + ylab('')+theme(legend.position = 'none',axis.text.y = element_text()) + scale_y_discrete(position = "right")
ggsave('~/Dropbox (Mason Lab)/i4/immune_modeling/impactfulbugplot.pdf',width=6,height=6)


ggplot(impactfulbugplot %>% filter(dset == 'GENE-CATALOG'),aes(x = n,y =microbialfeaturename,fill = celltype))+ geom_bar(stat='identity') + facet_grid(dset +regressionclass~ seqtype  ,scales = 'free_y',space = 'free_y') + theme_minimal() + scale_fill_brewer(palette = 'Set3') + xlab('Count') + ylab('')+theme(legend.position = 'none')+ scale_y_discrete(position = "right")
ggsave('~/Dropbox (Mason Lab)/i4/immune_modeling/impactfulbugplot_gc.pdf',width=9,height=8.2)


ggplot(impactfulbugplot %>% filter(dset != 'Bacterial -- METAPHLAN4',regressionclass == 'Skin'),aes(x = n,y =microbialfeaturename,fill = celltype))+ geom_bar(stat='identity') + facet_grid(dset +regressionclass~ seqtype  ,scales = 'free_y',space = 'free_y') + theme_minimal() + scale_fill_brewer(palette = 'Set3') + xlab('Count') + ylab('')+theme(legend.position = 'bottom')
ggsave('~/Dropbox (Mason Lab)/i4/immune_modeling/impactfulbugplot_skin.pdf',width=10,height=14)

#### GET MOST IMPACTED HUMAN GENES BY CELL TYPE

tophits = alldata %>% filter(dset != 'GENE-CATALOG')%>% ungroup %>% filter(features!="0")%>% select(celltype,coefficient,features,dset) %>% group_by(celltype,dset) %>% filter(coefficient>0) %>% slice_max(coefficient,n=5) 
bottomhits = alldata %>% filter(dset != 'GENE-CATALOG')%>% ungroup%>% filter(features!="0") %>% select(celltype,coefficient,features,dset) %>% group_by(celltype,dset) %>% filter(coefficient<0) %>% slice_min(coefficient,n=5)

allhits = bind_rows(tophits,bottomhits) %>% arrange(coefficient) %>% group_by(celltype) %>% arrange(desc(coefficient)) 
library(annotables)
annos = melt(grch38 %>% select(-start,-end,-strand),c('description')) 

allhits = left_join(allhits,annos %>% distinct,by=c('features'='value')) %>% mutate(label = if_else(is.na(description),features,description))

ggsave('~/Dropbox (Mason Lab)/i4/immune_modeling/tophumangenes_annotations_BAC.pdf',width=7,height=12)
2
ggplot(allhits %>% filter(dset == 'VIRAL'),aes(x = coefficient, y = reorder(label,coefficient))) + geom_bar(stat='identity') + facet_grid(celltype ~.  ,scales = 'free_y',space = 'free_y') + theme_bw() + xlab('Coefficient') + ylab('') + ggtitle('Top virus-associated human genes')
ggsave('~/Dropbox (Mason Lab)/i4/immune_modeling/tophumangenes_annotations_VIR.pdf',width=6,height=6)

ggplot(allhits,aes(x = coefficient, y = reorder(label,coefficient))) + geom_bar(stat='identity') + facet_grid(dset + celltype ~.  ,scales = 'free_y',space = 'free_y') + theme_bw() + xlab('Coefficient') + ylab('') + ggtitle('Top associated human genes')
ggsave('~/Dropbox (Mason Lab)/i4/immune_modeling/tophumangenes_annotation.pdf',width=6,height=15)



```
####### ICC ANALYSIS
```{r}

setwd('~/Dropbox (Mason Lab)/i4/')

# PARSE METADATA
metasub = meta %>% filter(Body.Location != "Swab Water",Body.Location != 'Open Air Control', Body.Location != "Deltoid - Pre-Biospy", !is.na(Body.Location)) %>% mutate(iscapsule = if_else(location == 'Capsule',1,0))  %>% mutate(skinoralnasal = if_else(isoral == 1,'ORAL',if_else(isskin==1,'SKIN','NASAL'))) %>% mutate(skinoralnasalcap = if_else(iscapsule == 1,'CAPSULE',skinoralnasal))
metasub$Timepoint_Recode = factor(metasub$Timepoint_Recode,levels = c('PRE-LAUNCH','MID-FLIGHT','POST-LAUNCH'))
metasub = metasub %>% mutate(Timepoint = if_else(Timepoint == 'Flight 1' | Timepoint == 'Flight 2','Mid-Flight',as.character(Timepoint)))
metasub$Timepoint = factor(metasub$Timepoint,levels=c('21-Jun','21-Aug','Sept Pre-Launch','Mid-Flight','Sept Post-Return','November','21-Dec',NA))
metasub$Timepoint_Recode = factor(metasub$Timepoint_Recode,levels=c('MID-FLIGHT','PRE-LAUNCH','POST-LAUNCH'))

build_icc_dfs <- function(regression_output2,c,regclass,st,dataset,abdata,metasub){
  sigs = c %>% filter(dset == dataset,seqtype == st,regclass2!='Overall',regclass2!='OVERALL') %>% filter(!grepl('-- LP',timetrend),!grepl('Decrease',timetrend)) %>% ungroup %>% select(yvar,regclass2) %>% distinct %>% mutate(regressionclass = if_else(regclass2 == 'Oral','Oral',if_else(regclass2=='Nasal','Nasal','Skin'))) 
  minval = min(abdata[abdata>0])
  abdata_meta = inner_join(abdata %>% rownames_to_column('SeqID'),metasub,by='SeqID') 
  relsigs = sigs %>% filter(regressionclass == regclass) %>% select(yvar) %>% unlist %>% unname%>% unique
  iccs = regression_output2 %>% filter(dset == dataset,seqtype == st,regressionclass == regclass) %>% filter(yvar %in% relsigs) %>% filter(effect == 'ran_pars') %>% select(group,estimate,yvar,regressionclass) %>% group_by(yvar) %>% summarise(group=group,total = sum(estimate),exp_crew = estimate/total) %>% ungroup%>% filter(group == 'Crew.ID') %>% mutate(loc2 = 'ICC')%>% select(loc2, yvar,exp_crew)%>% arrange(exp_crew) %>% rename(value = exp_crew) %>% mutate(regressionclass = regclass)
  iccs$yvar = fct_reorder(factor(iccs$yvar),iccs$value)
  prevs_melt = abdata_meta  %>% select(Crew.ID,Timepoint_Recode,location,any_of(relsigs)) %>% melt
  prevs_melt1 = prevs_melt %>% group_by(Crew.ID,Timepoint_Recode,location,variable) %>% summarise(average_ab = mean(value))
  totals =  prevs_melt %>% group_by(Crew.ID,location,Timepoint_Recode,variable) %>% summarise(totals = n())
  prevs_melt2 = prevs_melt %>% filter(value > 0) %>% group_by(Crew.ID,location,variable) %>% summarise(prevalence = n())
  prevs_melt = inner_join(prevs_melt1,prevs_melt2) 
  prevs_melt = inner_join(prevs_melt,totals) %>% mutate(prevalence = prevalence/totals) %>% select(-totals) %>% select(Crew.ID,variable,prevalence,average_ab,location,Timepoint_Recode) %>% dplyr::rename(yvar = variable) %>% melt(id.vars = c("Crew.ID","yvar","location","Timepoint_Recode")) %>% rename(loc2 = Crew.ID) %>% mutate(regressionclass = if_else(location != 'Oral' & location != 'Capsule' & location != 'Nasal','Skin',location)) %>% filter(regressionclass == regclass | regressionclass == 'Capsule')
  prevs_melt = prevs_melt %>% filter(yvar %in% relsigs)
  return(list(iccs,prevs_melt))
}

regression_output2 = regression_output %>% filter(dset == 'GTDB')

oraliccprevs = build_icc_dfs(regression_output2,categories,'Oral','METAGENOMICS','GTDB',gtdb_metag %>% t %>% as.data.frame(check.names=F),metasub)
#nasaliccprevs = build_icc_dfs(regression_output2,sigs,'Nasal','METAGENOMICS','GTDB',abdata_meta)
skiniccprevs = build_icc_dfs(regression_output2,c,'Skin','METAGENOMICS','GTDB',gtdb_metag%>% t %>% as.data.frame(check.names=F),metasub)

iccs = bind_rows(oraliccprevs[[1]]%>% mutate(associationtype = 'Oral'),skiniccprevs[[1]]%>% mutate(associationtype = 'Skin') )
prevs = bind_rows(oraliccprevs[[2]]%>% mutate(associationtype = 'Oral'),skiniccprevs[[2]]%>% mutate(associationtype = 'Skin'))
##### ABOVE SHOULD BE ENOUGH TO GENERATE PLOT SANS PHYLUM ANNOTATION, WHICH WILL BE A BARPLOT
iccs$yvar = fct_reorder(factor(iccs$yvar),iccs$value)
prevs$yvar = factor(prevs$yvar,levels(iccs$yvar))

prevs$value [prevs$value ==0] =NA

colors = setNames(as.list(brewer.pal('Set3',n=12)),unique(metasub$location))

plot1 = ggplot(data = iccs,aes(x=yvar,y=value)) +theme_cowplot()+ geom_point(stat='identity',alpha=.7,size=2)  + theme(axis.text.x = element_blank())  + theme(legend.position = 'top',axis.title.x = element_blank())+ facet_grid(.~associationtype,scales='free_x',space='free_x') + ylab('ICC')

foo =  prevs %>% filter(variable =='average_ab',Timepoint_Recode == 'MID-FLIGHT') %>% filter(!is.na(value))

foo = foo %>% mutate(labelname = if_else(value>0 & variable =='average_ab' &location == 'Capsule',strsplit(as.character(yvar),'s__') %>% map_chr(2),''))


minval = 0.00000001

plot2 = ggplot(data =foo,aes(x=yvar,y=(value+minval),color=location))+theme_cowplot() + geom_point(stat='identity',alpha=.7,size=2) + facet_grid(loc2 ~ associationtype,scales='free_x',space='free_x') + theme(axis.text.x = element_blank(),axis.title.x = element_blank()) + scale_y_log10() + scale_color_manual(values = colors) + theme(legend.position = 'bottom') + ylab('log10(Relative Abundance)') 


plot1/plot2

ggsave('~/Dropbox (Mason Lab)/i4/iccanalysis/icc_bacterial_metag.pdf',width=10,height=10)
iccs_gtdb_metag  = iccs

```
####### ENV VS CREW ACQUISITION -- BACTERIAL
```{r}
abdata_meta = inner_join(gtdb_metag %>% rownames_to_column('SeqID'),metasub %>% filter(!(location == 'Capsule' & Timepoint_Recode == 'PRE-FLIGHT'))%>% select(SeqID,Timepoint,Timepoint_Recode,Crew.ID,location),by='SeqID')  %>% melt %>% select(-SeqID) %>% mutate(skinoralnasalcapsule = if_else(location == 'Oral' | location == "Capsule" | location == 'Nasal',location,'Skin')) 
prevalence_across_time = abdata_meta %>% group_by(variable,skinoralnasalcapsule,location,Crew.ID,Timepoint) %>% summarize(n = mean(value > 0)) %>% dcast(Crew.ID + variable + skinoralnasalcapsule + location~ Timepoint, value.var = "n")
cap = prevalence_across_time %>% filter(location == 'Capsule') %>% select(variable,"Mid-Flight") %>% rename(capsule_prev_midflight = `Mid-Flight`)

prevalence_across_time = prevalence_across_time %>% group_by(variable,skinoralnasalcapsule,location) %>% summarise(pre_launch = sum(`Sept Pre-Launch`>0),post_launch = sum(`Sept Post-Return`>0))

prevalence_across_time = left_join(prevalence_across_time %>% filter(location != 'Capsule'),cap)
prevalence_across_time[is.na(prevalence_across_time)] = 0 

prevalence_across_time$capsule_prev_midflight[prevalence_across_time$capsule_prev_midflight>0]=1
prevalence_across_time$names = paste(prevalence_across_time$variable,prevalence_across_time$location,prevalence_across_time$skinoralnasalcapsule)

prevalence_across_time = prevalence_across_time %>% group_by(location) %>% arrange(location,desc(capsule_prev_midflight),pre_launch,post_launch) %>% column_to_rownames('names') %>% select(pre_launch,capsule_prev_midflight,post_launch,location) %>% filter(post_launch > pre_launch,post_launch>1)

temp = prevalence_across_time %>% select(location,capsule_prev_midflight) %>% group_by(location) %>% summarise(mean = mean(capsule_prev_midflight)) %>% arrange(desc(mean)) %>% select(location) %>% unlist %>% unname

prevalence_across_time$location=factor(prevalence_across_time$location,levels=temp)

prevalence_across_time = prevalence_across_time  %>% arrange(location,desc(capsule_prev_midflight),pre_launch,post_launch) 

col1 = c('1'='red','2'='blue','3'='cyan','4'='green','0' = 'black')
col2 = c('0' = 'black','1'='grey')

h1 = Heatmap(prevalence_across_time %>% ungroup %>% select(pre_launch) %>% t %>% as.data.frame,column_split = prevalence_across_time$location,show_row_names = T,show_column_names = F,cluster_columns =F,cluster_rows=F,col=col1,column_title_rot = 90)

h2 = Heatmap(prevalence_across_time %>% ungroup %>% select(capsule_prev_midflight) %>% t %>% as.data.frame,column_split = prevalence_across_time$location,show_row_names = T,show_column_names = F,cluster_columns =F,cluster_rows=F,col=col2)

h3 = Heatmap(prevalence_across_time %>% ungroup %>% select(post_launch) %>% t %>% as.data.frame,column_split = prevalence_across_time$location,show_row_names = T,show_column_names = F,cluster_columns =F,cluster_rows=F,col=col1,column_title_rot = 90)

pdf('~/Dropbox (Mason Lab)/i4/iccanalysis/shared_env_signal_bacmetag.pdf',width=8,height=3)
h1 %v% h2 %v% h3
dev.off()

prop_shared_bacmetag = prevalence_across_time %>% select(location,capsule_prev_midflight) %>% group_by(location) %>% summarise(shared_with_environment = mean(capsule_prev_midflight)) %>% mutate(shared_between_crew = 1-shared_with_environment,dset = 'Bacterial/Archaeal') %>% melt
```
####### ENV VS CREW ACQUISITION -- VIRAL
```{r}
abdata_meta = inner_join(viral_metat %>% t %>% as.data.frame(check.names=F) %>% rownames_to_column('SeqID'),metasub %>% filter(!(location == 'Capsule' & Timepoint_Recode == 'PRE-FLIGHT'))%>% select(SeqID,Timepoint,Timepoint_Recode,Crew.ID,location),by='SeqID')  %>% melt %>% select(-SeqID) %>% mutate(skinoralnasalcapsule = if_else(location == 'Oral' | location == "Capsule" | location == 'Nasal',location,'Skin')) 
prevalence_across_time = abdata_meta %>% group_by(variable,skinoralnasalcapsule,location,Crew.ID,Timepoint) %>% summarize(n = mean(value > 0)) %>% dcast(Crew.ID + variable + skinoralnasalcapsule + location~ Timepoint, value.var = "n")
cap = prevalence_across_time %>% filter(location == 'Capsule') %>% select(variable,"Mid-Flight") %>% rename(capsule_prev_midflight = `Mid-Flight`)

prevalence_across_time = prevalence_across_time %>% group_by(variable,skinoralnasalcapsule,location) %>% summarise(pre_launch = sum(`Sept Pre-Launch`>0),post_launch = sum(`Sept Post-Return`>0))

prevalence_across_time = left_join(prevalence_across_time %>% filter(location != 'Capsule'),cap)
prevalence_across_time[is.na(prevalence_across_time)] = 0 

prevalence_across_time$capsule_prev_midflight[prevalence_across_time$capsule_prev_midflight>0]=1
prevalence_across_time$names = paste(prevalence_across_time$variable,prevalence_across_time$location,prevalence_across_time$skinoralnasalcapsule)

prevalence_across_time = prevalence_across_time %>% group_by(location) %>% arrange(location,desc(capsule_prev_midflight),pre_launch,post_launch) %>% column_to_rownames('names') %>% select(pre_launch,capsule_prev_midflight,post_launch,location)%>% filter(post_launch > pre_launch,post_launch>1)

temp = prevalence_across_time %>% select(location,capsule_prev_midflight) %>% group_by(location) %>% summarise(mean = mean(capsule_prev_midflight)) %>% arrange(desc(mean)) %>% select(location) %>% unlist %>% unname

prevalence_across_time$location=factor(prevalence_across_time$location,levels=temp)

prevalence_across_time = prevalence_across_time  %>% arrange(location,desc(capsule_prev_midflight),pre_launch,post_launch) 

col1 = c('1'='red','2'='blue','3'='cyan','4'='green','0' = 'black')
col2 = c('0' = 'black','1'='grey')

h1 = Heatmap(prevalence_across_time %>% ungroup %>% select(pre_launch) %>% t %>% as.data.frame,column_split = prevalence_across_time$location,show_row_names = T,show_column_names = F,cluster_columns =F,cluster_rows=F,col=col1,column_title_rot = 90)

h2 = Heatmap(prevalence_across_time %>% ungroup %>% select(capsule_prev_midflight) %>% t %>% as.data.frame,column_split = prevalence_across_time$location,show_row_names = T,show_column_names = F,cluster_columns =F,cluster_rows=F,col=col2)

h3 = Heatmap(prevalence_across_time %>% ungroup %>% select(post_launch) %>% t %>% as.data.frame,column_split = prevalence_across_time$location,show_row_names = T,show_column_names = F,cluster_columns =F,cluster_rows=F,col=col1,column_title_rot = 90)

pdf('~/Dropbox (Mason Lab)/i4/iccanalysis/shared_env_signal_virmetat.pdf',width=8,height=3)
h1 %v% h2 %v% h3
dev.off()

prop_shared_virmetat = prevalence_across_time %>% select(location,capsule_prev_midflight) %>% group_by(location) %>% summarise(shared_with_environment = mean(capsule_prev_midflight)) %>% mutate(shared_between_crew = 1-shared_with_environment,dset = 'Viral') %>% melt
```
####### ENV VS CREW ACQUISITION -- SUPP
```{r}
# env shared summary

envsharedplot = bind_rows(prop_shared_bacmetag,prop_shared_virmetat) %>% mutate(skinoralnasal = if_else(location != 'Oral' & location != 'Nasal','Skin',as.character(location)))
ggplot(data = envsharedplot,aes(x = location,y = value*100,fill = variable)) + geom_bar(stat='identity') +facet_grid(dset ~ skinoralnasal,scales='free',space='free') + theme_bw() + theme(axis.text.x = element_text(angle=60,hjust=1)) + xlab('') + ylab('%') + scale_fill_manual(values=c('gray','black'))
ggsave('~/Dropbox (Mason Lab)/i4/iccanalysis/shared_env_signal_virmetat_overall.pdf',width=6,height=5)

envsharedplot$skinoralnasal = factor(envsharedplot$skinoralnasal,levels = c('Oral','Nasal','Skin'))

ggplot(envsharedplot %>% filter(variable == 'shared_with_environment') %>% group_by(skinoralnasal,dset)%>% summarise(mean = mean(value)*100,sd = sd(value)*100),aes(x = skinoralnasal,y=mean,fill=dset)) + geom_bar(stat='identity',position='dodge') + scale_fill_manual(values=c('lightgray','black')) + xlab('') + ggtitle('Taxa potentially propagated through capsule') + ylab('%') + geom_errorbar(aes(ymin = mean -sd,ymax = mean+sd), width=.2, position=position_dodge(.9),color='darkgrey') + theme_bw() + geom_text(aes(label=round(mean,1)), position=position_dodge(width=0.9), vjust=-0.25)
ggsave('~/Dropbox (Mason Lab)/i4/iccanalysis/propagation_rates.pdf',width=6,height=5)

```
####### ICC PLOTS FOR SUPP
```{r}

#### FOR SUPP

oraliccprevs = build_icc_dfs(regression_output2,c,'Oral','METAGENOMICS','ASSEMBLED-BACTERIA',bass_metag,metasub)
#nasaliccprevs = build_icc_dfs(regression_output2,sigs,'Nasal','METAGENOMICS','GTDB',abdata_meta)
skiniccprevs = build_icc_dfs(regression_output2,c,'Skin','METAGENOMICS','ASSEMBLED-BACTERIA',bass_metag,metasub)

iccs = bind_rows(oraliccprevs[[1]]%>% mutate(associationtype = 'Oral'),skiniccprevs[[1]]%>% mutate(associationtype = 'Skin') )
prevs = bind_rows(oraliccprevs[[2]]%>% mutate(associationtype = 'Oral'),skiniccprevs[[2]]%>% mutate(associationtype = 'Skin'))
##### ABOVE SHOULD BE ENOUGH TO GENERATE PLOT SANS PHYLUM ANNOTATION, WHICH WILL BE A BARPLOT
iccs$yvar = fct_reorder(factor(iccs$yvar),iccs$value)
prevs$yvar = factor(prevs$yvar,levels(iccs$yvar))

plot1 = ggplot(data = iccs,aes(x=yvar,y=value)) + geom_point(stat='identity',alpha=.7,size=2)  + theme(axis.text.x = element_blank())  + theme(legend.position = 'top',axis.title.x = element_blank())+ facet_grid(.~associationtype,scales='free_x',space='free_x') + ylab('ICC')

prevs$value [prevs$value ==0] =NA

plot2 = ggplot(data = prevs %>% filter(variable =='average_ab',Timepoint_Recode == 'MID-FLIGHT',regressionclass!='NASAL') ,aes(x=yvar,y=(value+minval),color=location)) + geom_point(stat='identity',alpha=.7,size=2) + facet_grid(loc2 ~ associationtype,scales='free_x',space='free_x') + theme(axis.text.x = element_blank(),axis.title.x = element_blank()) + scale_y_log10() + scale_color_brewer(palette = 'Set3') + theme(legend.position = 'Bottom') + ylab('log10(Relative Abundance)')

plot1/plot2

ggsave('~/Dropbox (Mason Lab)/i4/iccanalysis/icc_assembledbacterial_metag.pdf',width=8,height=10)

oraliccprevs = build_icc_dfs(regression_output2,c,'Oral','METATRANSCRIPTOMICS','ASSEMBLED-VIRUSES',viral_ass_metat %>% t %>% as.data.frame(check.names=F) ,metasub)
#nasaliccprevs = build_icc_dfs(regression_output2,c,'Nasal','METATRANSCRIPTOMICS','ASSEMBLED-VIRUSES',viral_ass_metat %>% t %>% as.data.frame(check.names=F) ,metasub)
skiniccprevs = build_icc_dfs(regression_output2,c,'Skin','METATRANSCRIPTOMICS','ASSEMBLED-VIRUSES',viral_ass_metat %>% t %>% as.data.frame(check.names=F) ,metasub)

iccs = bind_rows(oraliccprevs[[1]]%>% mutate(associationtype = 'Oral'),skiniccprevs[[1]]%>% mutate(associationtype = 'Skin'),nasaliccprevs[[1]]%>% mutate(associationtype = 'Nasal') )
prevs = bind_rows(oraliccprevs[[2]]%>% mutate(associationtype = 'Oral'),skiniccprevs[[2]]%>% mutate(associationtype = 'Skin'),nasaliccprevs[[2]]%>% mutate(associationtype = 'Nasal'))
##### ABOVE SHOULD BE ENOUGH TO GENERATE PLOT SANS PHYLUM ANNOTATION, WHICH WILL BE A BARPLOT
iccs$yvar = fct_reorder(factor(iccs$yvar),iccs$value)
prevs$yvar = factor(prevs$yvar,levels(iccs$yvar))

plot3 = ggplot(data = iccs,aes(x=yvar,y=value)) + geom_point(stat='identity',alpha=.7,size=2)  + theme(axis.text.x = element_blank())  + theme(legend.position = 'top',axis.title.x = element_blank())+ facet_grid(.~associationtype,scales='free_x',space='free_x') + ylab('ICC')

plot4 = ggplot(data = prevs %>% filter(variable =='average_ab',Timepoint_Recode == 'MID-FLIGHT',regressionclass!='NASAL'),aes(x=yvar,y=(value+minval),color=location)) + geom_point(stat='identity',alpha=.7,size=2) + facet_grid(loc2 ~ associationtype,scales='free_x',space='free_x') + theme(axis.text.x = element_blank(),axis.title.x = element_blank()) + scale_y_log10() + scale_color_brewer(palette = 'Set3') + theme(legend.position = 'bottom') + ylab('log10(Relative Abundance)')

plot3/plot4
ggsave('~/Dropbox (Mason Lab)/i4/iccanalysis/icc_viral_metat_ass.pdf',width=8,height=10)

```

