
### KRAKEN2

## NO CONF NO MASK 

Rscript decontaminate_and_clean_files.R bac-vir-fung species kraken2_bracken_abundances/i4_wgs_mtx_bracken_noconf_nomask_species.tsv kraken2 nomask-noconf metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_noconf_nomask.tsv 

Rscript decontaminate_and_clean_files.R bac-vir-fung genus kraken2_bracken_abundances/i4_wgs_mtx_bracken_noconf_nomask_genus.tsv kraken2 nomask-noconf metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_noconf_nomask_genus.tsv 

Rscript decontaminate_and_clean_files.R bac-vir-fung family kraken2_bracken_abundances/i4_wgs_mtx_bracken_noconf_nomask_family.tsv kraken2 nomask-noconf metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_noconf_nomask_family.tsv 

Rscript decontaminate_and_clean_files.R bac-vir-fung order kraken2_bracken_abundances/i4_wgs_mtx_bracken_noconf_nomask_order.tsv kraken2 nomask-noconf metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_noconf_nomask_order.tsv 

Rscript decontaminate_and_clean_files.R bac-vir-fung class kraken2_bracken_abundances/i4_wgs_mtx_bracken_noconf_nomask_class.tsv kraken2 nomask-noconf metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_noconf_nomask_class.tsv 

Rscript decontaminate_and_clean_files.R bac-vir-fung phylum kraken2_bracken_abundances/i4_wgs_mtx_bracken_noconf_nomask_phylum.tsv kraken2 nomask-noconf metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_noconf_nomask_phylum.tsv


## NO CONF MASK

Rscript decontaminate_and_clean_files.R bac-vir-fung species kraken2_bracken_abundances/i4_wgs_mtx_bracken_noconf_mask_species.tsv kraken2 mask-noconf metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_noconf_mask.tsv  

Rscript decontaminate_and_clean_files.R bac-vir-fung genus kraken2_bracken_abundances/i4_wgs_mtx_bracken_noconf_mask_genus.tsv kraken2 mask-noconf metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_noconf_mask_genus.tsv 

Rscript decontaminate_and_clean_files.R bac-vir-fung family kraken2_bracken_abundances/i4_wgs_mtx_bracken_noconf_mask_family.tsv kraken2 mask-noconf metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_noconf_mask_family.tsv 

Rscript decontaminate_and_clean_files.R bac-vir-fung order kraken2_bracken_abundances/i4_wgs_mtx_bracken_noconf_mask_order.tsv kraken2 mask-noconf metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_noconf_mask_order.tsv  

Rscript decontaminate_and_clean_files.R bac-vir-fung class kraken2_bracken_abundances/i4_wgs_mtx_bracken_noconf_mask_class.tsv kraken2 mask-noconf metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_noconf_mask_class.tsv 

Rscript decontaminate_and_clean_files.R bac-vir-fung phylum kraken2_bracken_abundances/i4_wgs_mtx_bracken_noconf_mask_phylum.tsv kraken2 mask-noconf metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_noconf_mask_phylum.tsv


## CONF NO MASK

Rscript decontaminate_and_clean_files.R bac-vir-fung species kraken2_bracken_abundances/i4_wgs_mtx_bracken_conf_nomask_species.tsv kraken2 nomask-conf metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_conf_nomask.tsv  

Rscript decontaminate_and_clean_files.R bac-vir-fung genus kraken2_bracken_abundances/i4_wgs_mtx_bracken_conf_nomask_genus.tsv kraken2 nomask-conf metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_conf_nomask_genus.tsv 

Rscript decontaminate_and_clean_files.R bac-vir-fung family kraken2_bracken_abundances/i4_wgs_mtx_bracken_conf_nomask_family.tsv kraken2 nomask-conf metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_conf_nomask_family.tsv 

Rscript decontaminate_and_clean_files.R bac-vir-fung order kraken2_bracken_abundances/i4_wgs_mtx_bracken_conf_nomask_order.tsv kraken2 nomask-conf metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_conf_nomask_order.tsv 

Rscript decontaminate_and_clean_files.R bac-vir-fung class kraken2_bracken_abundances/i4_wgs_mtx_bracken_conf_nomask_class.tsv kraken2 nomask-conf metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_conf_nomask_class.tsv 

Rscript decontaminate_and_clean_files.R bac-vir-fung phylum kraken2_bracken_abundances/i4_wgs_mtx_bracken_conf_nomask_phylum.tsv kraken2 nomask-conf metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_conf_nomask_phylum.tsv


## CONF MASK

Rscript decontaminate_and_clean_files.R bac-vir-fung species kraken2_bracken_abundances/i4_wgs_mtx_bracken_conf_mask_species.tsv kraken2 conf-mask metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_conf_mask.tsv 

Rscript decontaminate_and_clean_files.R bac-vir-fung genus kraken2_bracken_abundances/i4_wgs_mtx_bracken_conf_mask_genus.tsv kraken2 conf-mask metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_conf_mask_genus.tsv 

Rscript decontaminate_and_clean_files.R bac-vir-fung family kraken2_bracken_abundances/i4_wgs_mtx_bracken_conf_mask_family.tsv kraken2 conf-mask metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_conf_mask_family.tsv 

Rscript decontaminate_and_clean_files.R bac-vir-fung order kraken2_bracken_abundances/i4_wgs_mtx_bracken_conf_mask_order.tsv kraken2 conf-mask metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_conf_mask_order.tsv 

Rscript decontaminate_and_clean_files.R bac-vir-fung class kraken2_bracken_abundances/i4_wgs_mtx_bracken_conf_mask_class.tsv kraken2 conf-mask metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_conf_mask_class.tsv 

Rscript decontaminate_and_clean_files.R bac-vir-fung phylum kraken2_bracken_abundances/i4_wgs_mtx_bracken_conf_mask_phylum.tsv kraken2 conf-mask metasub_negatives/abundance_tables/i4_wgs_mtx_bracken_conf_mask_phylum.tsv


### METAPHLAN 4
Rscript decontaminate_and_clean_files.R bacteria t__ metaphlan4/metaphlan_i4.tsv metaphlan4 default metasub_negatives/abundance_tables/metasub_negatives_metaphlan.tsv 

Rscript decontaminate_and_clean_files.R bacteria s__ metaphlan4/metaphlan_i4.tsv metaphlan4 default metasub_negatives/abundance_tables/metasub_negatives_metaphlan.tsv 

Rscript decontaminate_and_clean_files.R bacteria g__ metaphlan4/metaphlan_i4.tsv metaphlan4 default metasub_negatives/abundance_tables/metasub_negatives_metaphlan.tsv 

Rscript decontaminate_and_clean_files.R bacteria f__ metaphlan4/metaphlan_i4.tsv metaphlan4 default metasub_negatives/abundance_tables/metasub_negatives_metaphlan.tsv 

Rscript decontaminate_and_clean_files.R bacteria o__ metaphlan4/metaphlan_i4.tsv metaphlan4 default metasub_negatives/abundance_tables/metasub_negatives_metaphlan.tsv 

Rscript decontaminate_and_clean_files.R bacteria c__ metaphlan4/metaphlan_i4.tsv metaphlan4 default metasub_negatives/abundance_tables/metasub_negatives_metaphlan.tsv 

Rscript decontaminate_and_clean_files.R bacteria p__ metaphlan4/metaphlan_i4.tsv metaphlan4 default metasub_negatives/abundance_tables/metasub_negatives_metaphlan.tsv 

### XTREE -- GTDB 
Rscript decontaminate_and_clean_files.R bacteria s__ xtree_merged/GTDB_.05_.0025_s_metagenomics_ra.tsv,xtree_merged/GTDB_.05_.0025_s_metatranscriptomics_ra.tsv xtree 05-0025 metasub_negatives/abundance_tables/GTDB_.05_.0025_s_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R bacteria g__ xtree_merged/GTDB_.05_.0025_g_metagenomics_ra.tsv,xtree_merged/GTDB_.05_.0025_g_metatranscriptomics_ra.tsv xtree 05-0025 metasub_negatives/abundance_tables/GTDB_.05_.0025_g_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R bacteria f__ xtree_merged/GTDB_.05_.0025_f_metagenomics_ra.tsv,xtree_merged/GTDB_.05_.0025_f_metatranscriptomics_ra.tsv xtree 05-0025 metasub_negatives/abundance_tables/GTDB_.05_.0025_f_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R bacteria c__ xtree_merged/GTDB_.05_.0025_c_metagenomics_ra.tsv,xtree_merged/GTDB_.05_.0025_c_metatranscriptomics_ra.tsv xtree 05-0025 metasub_negatives/abundance_tables/GTDB_.05_.0025_c_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R bacteria o__ xtree_merged/GTDB_.05_.0025_o_metagenomics_ra.tsv,xtree_merged/GTDB_.05_.0025_o_metatranscriptomics_ra.tsv xtree 05-0025 metasub_negatives/abundance_tables/GTDB_.05_.0025_o_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R bacteria p__ xtree_merged/GTDB_.05_.0025_p_metagenomics_ra.tsv,xtree_merged/GTDB_.05_.0025_p_metatranscriptomics_ra.tsv xtree 05-0025 metasub_negatives/abundance_tables/GTDB_.05_.0025_p_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R bacteria s__ xtree_merged/GTDB_.005_.0025_s_metagenomics_ra.tsv,xtree_merged/GTDB_.005_.0025_s_metatranscriptomics_ra.tsv xtree 005-0025 metasub_negatives/abundance_tables/GTDB_.005_.0025_s_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R bacteria g__ xtree_merged/GTDB_.005_.0025_g_metagenomics_ra.tsv,xtree_merged/GTDB_.005_.0025_g_metatranscriptomics_ra.tsv xtree 005-0025 metasub_negatives/abundance_tables/GTDB_.005_.0025_g_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R bacteria f__ xtree_merged/GTDB_.005_.0025_f_metagenomics_ra.tsv,xtree_merged/GTDB_.005_.0025_f_metatranscriptomics_ra.tsv xtree 005-0025 metasub_negatives/abundance_tables/GTDB_.005_.0025_f_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R bacteria c__ xtree_merged/GTDB_.005_.0025_c_metagenomics_ra.tsv,xtree_merged/GTDB_.005_.0025_c_metatranscriptomics_ra.tsv xtree 005-0025 metasub_negatives/abundance_tables/GTDB_.005_.0025_c_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R bacteria o__ xtree_merged/GTDB_.005_.0025_o_metagenomics_ra.tsv,xtree_merged/GTDB_.005_.0025_o_metatranscriptomics_ra.tsv xtree 005-0025 metasub_negatives/abundance_tables/GTDB_.005_.0025_o_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R bacteria p__ xtree_merged/GTDB_.005_.0025_p_metagenomics_ra.tsv,xtree_merged/GTDB_.005_.0025_p_metatranscriptomics_ra.tsv xtree 005-0025 metasub_negatives/abundance_tables/GTDB_.005_.0025_p_metagenomics_ra.tsv


### VIRAL -- low cutoff

Rscript decontaminate_and_clean_files.R virus s__ xtree_merged/genbank-viral_.01_.0.005_metagenomics_species_ra.tsv,xtree_merged/genbank-viral_.01_.0.005_metatranscriptomics_species_ra.tsv xtree 01-005 metasub_negatives/abundance_tables/genbank-viral_.01_.0.005_metagenomics_species_ra.tsv

Rscript decontaminate_and_clean_files.R virus g__ xtree_merged/genbank-viral_.01_.0.005_g_metagenomics_ra.tsv,xtree_merged/genbank-viral_.01_.0.005_g_metatranscriptomics_ra.tsv xtree 01-005 metasub_negatives/abundance_tables/genbank-viral_.01_.0.005_g_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R virus f__ xtree_merged/genbank-viral_.01_.0.005_f_metagenomics_ra.tsv,xtree_merged/genbank-viral_.01_.0.005_f_metatranscriptomics_ra.tsv xtree 01-005 metasub_negatives/abundance_tables/genbank-viral_.01_.0.005_f_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R virus o__ xtree_merged/genbank-viral_.01_.0.005_o_metagenomics_ra.tsv,xtree_merged/genbank-viral_.01_.0.005_o_metatranscriptomics_ra.tsv xtree 01-005 metasub_negatives/abundance_tables/genbank-viral_.01_.0.005_o_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R virus c__ xtree_merged/genbank-viral_.01_.0.005_c_metagenomics_ra.tsv,xtree_merged/genbank-viral_.01_.0.005_c_metatranscriptomics_ra.tsv xtree 01-005 metasub_negatives/abundance_tables/genbank-viral_.01_.0.005_c_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R virus p__ xtree_merged/genbank-viral_.01_.0.005_p_metagenomics_ra.tsv,xtree_merged/genbank-viral_.01_.0.005_p_metatranscriptomics_ra.tsv xtree 01-005 metasub_negatives/abundance_tables/genbank-viral_.01_.0.005_p_metagenomics_ra.tsv

### VIRAL -- high cutoff

Rscript decontaminate_and_clean_files.R virus s__ xtree_merged/genbank-viral_.1_.0.05_metagenomics_species_ra.tsv,xtree_merged/genbank-viral_.1_.0.05_metatranscriptomics_species_ra.tsv xtree 1-05 metasub_negatives/abundance_tables/genbank-viral_.1_.0.05_metagenomics_species_ra.tsv

Rscript decontaminate_and_clean_files.R virus g__ xtree_merged/genbank-viral_.1_.0.05_g_metagenomics_ra.tsv,xtree_merged/genbank-viral_.1_.0.05_g_metatranscriptomics_ra.tsv xtree 1-05 metasub_negatives/abundance_tables/genbank-viral_.1_.0.05_g_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R virus f__ xtree_merged/genbank-viral_.1_.0.05_f_metagenomics_ra.tsv,xtree_merged/genbank-viral_.1_.0.05_f_metatranscriptomics_ra.tsv xtree 1-05 metasub_negatives/abundance_tables/genbank-viral_.1_.0.05_f_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R virus o__ xtree_merged/genbank-viral_.1_.0.05_o_metagenomics_ra.tsv,xtree_merged/genbank-viral_.1_.0.05_o_metatranscriptomics_ra.tsv xtree 1-05 metasub_negatives/abundance_tables/genbank-viral_.1_.0.05_o_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R virus c__ xtree_merged/genbank-viral_.1_.0.05_c_metagenomics_ra.tsv,xtree_merged/genbank-viral_.1_.0.05_c_metatranscriptomics_ra.tsv xtree 1-05 metasub_negatives/abundance_tables/genbank-viral_.1_.0.05_c_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R virus p__ xtree_merged/genbank-viral_.1_.0.05_p_metagenomics_ra.tsv,xtree_merged/genbank-viral_.1_.0.05_p_metatranscriptomics_ra.tsv xtree 1-05 metasub_negatives/abundance_tables/genbank-viral_.1_.0.05_p_metagenomics_ra.tsv


### VIRAL -- medium cutoff

Rscript decontaminate_and_clean_files.R virus s__ xtree_merged/genbank-viral_.02_.0.015_metagenomics_species_ra.tsv,xtree_merged/genbank-viral_.02_.0.015_metatranscriptomics_species_ra.tsv xtree 02-015 metasub_negatives/abundance_tables/genbank-viral_.02_.0.015_metagenomics_species_ra.tsv

Rscript decontaminate_and_clean_files.R virus g__ xtree_merged/genbank-viral_.02_.0.015_g_metagenomics_ra.tsv,xtree_merged/genbank-viral_.02_.0.015_g_metatranscriptomics_ra.tsv xtree 02-015 metasub_negatives/abundance_tables/genbank-viral_.02_.0.015_g_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R virus f__ xtree_merged/genbank-viral_.02_.0.015_f_metagenomics_ra.tsv,xtree_merged/genbank-viral_.02_.0.015_f_metatranscriptomics_ra.tsv xtree 02-015 metasub_negatives/abundance_tables/genbank-viral_.02_.0.015_f_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R virus o__ xtree_merged/genbank-viral_.02_.0.015_o_metagenomics_ra.tsv,xtree_merged/genbank-viral_.02_.0.015_o_metatranscriptomics_ra.tsv xtree 02-015 metasub_negatives/abundance_tables/genbank-viral_.02_.0.015_o_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R virus c__ xtree_merged/genbank-viral_.02_.0.015_c_metagenomics_ra.tsv,xtree_merged/genbank-viral_.02_.0.015_c_metatranscriptomics_ra.tsv xtree 02-015 metasub_negatives/abundance_tables/genbank-viral_.02_.0.015_c_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R virus p__ xtree_merged/genbank-viral_.02_.0.015_p_metagenomics_ra.tsv,xtree_merged/genbank-viral_.02_.0.015_p_metatranscriptomics_ra.tsv xtree 02-015 metasub_negatives/abundance_tables/genbank-viral_.02_.0.015_p_metagenomics_ra.tsv


### VIRAL -- another cutoff

Rscript decontaminate_and_clean_files.R virus s__ xtree_merged/genbank-viral_.05_0.025_metagenomics_species_ra.tsv,xtree_merged/genbank-viral_.05_0.025_metatranscriptomics_species_ra.tsv xtree 05-025 metasub_negatives/abundance_tables/genbank-viral_.05_.0.025_metagenomics_species_ra.tsv

Rscript decontaminate_and_clean_files.R virus g__ xtree_merged/genbank-viral_.05_0.025_g_metagenomics_ra.tsv,xtree_merged/genbank-viral_.05_0.025_g_metatranscriptomics_ra.tsv xtree 05-025 metasub_negatives/abundance_tables/genbank-viral_.05_.0.025_g_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R virus f__ xtree_merged/genbank-viral_.05_0.025_f_metagenomics_ra.tsv,xtree_merged/genbank-viral_.05_0.025_f_metatranscriptomics_ra.tsv xtree 05-025 metasub_negatives/abundance_tables/genbank-viral_.05_.0.025_f_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R virus o__ xtree_merged/genbank-viral_.05_0.025_o_metagenomics_ra.tsv,xtree_merged/genbank-viral_.05_0.025_o_metatranscriptomics_ra.tsv xtree 05-025 metasub_negatives/abundance_tables/genbank-viral_.05_.0.025_o_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R virus c__ xtree_merged/genbank-viral_.05_0.025_c_metagenomics_ra.tsv,xtree_merged/genbank-viral_.05_0.025_c_metatranscriptomics_ra.tsv xtree 05-025 metasub_negatives/abundance_tables/genbank-viral_.05_.0.025_c_metagenomics_ra.tsv

Rscript decontaminate_and_clean_files.R virus p__ xtree_merged/genbank-viral_.05_0.025_p_metagenomics_ra.tsv,xtree_merged/genbank-viral_.05_0.025_p_metatranscriptomics_ra.tsv xtree 05-025 metasub_negatives/abundance_tables/genbank-viral_.05_.0.025_p_metagenomics_ra.tsv

### BACTERIAL MAGS

Rscript decontaminate_and_clean_files.R bacteria p__ MAGS/ASSEMBLED-BACTERIAL-MAGS_.05_.0025_metagenomics_species_ra.tsv,MAGS/ASSEMBLED-BACTERIAL-MAGS_.05_.0025_metatranscriptomics_species_ra.tsv MAG 05-0025 NONE

### VIRAL MAGS

Rscript decontaminate_and_clean_files.R virus p__ MAGS/ASSEMBLED-VIRAL-GENOMES_.01_.0.005_metagenomics_species_ra.tsv,MAGS/ASSEMBLED-VIRAL-GENOMES_.01_.0.005_metatranscriptomics_species_ra.tsv MAG 01-005 NONE

### PHANTA

Rscript decontaminate_and_clean_files.R virus species_ phanta/relative_taxonomic_abundance.txt phanta default metasub_negatives/abundance_tables/relative_taxonomic_abundance.txt

Rscript decontaminate_and_clean_files.R virus genus_ phanta/relative_taxonomic_abundance.txt phanta default metasub_negatives/abundance_tables/relative_taxonomic_abundance.txt

Rscript decontaminate_and_clean_files.R virus family_ phanta/relative_taxonomic_abundance.txt phanta default metasub_negatives/abundance_tables/relative_taxonomic_abundance.txt

Rscript decontaminate_and_clean_files.R virus order_ phanta/relative_taxonomic_abundance.txt phanta default metasub_negatives/abundance_tables/relative_taxonomic_abundance.txt

Rscript decontaminate_and_clean_files.R virus class_ phanta/relative_taxonomic_abundance.txt phanta default metasub_negatives/abundance_tables/relative_taxonomic_abundance.txt

Rscript decontaminate_and_clean_files.R virus phylum_ phanta/relative_taxonomic_abundance.txt phanta default metasub_negatives/abundance_tables/relative_taxonomic_abundance.txt
